{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a193039e-72a4-4d0a-9e14-b3c77dedf892",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "from collections import defaultdict\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.model_selection import KFold\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.metrics import accuracy_score, roc_curve, roc_auc_score, confusion_matrix, ConfusionMatrixDisplay, RocCurveDisplay\n",
    "from sklearn.preprocessing import StandardScaler, OneHotEncoder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "134dcc50-5e0d-4057-8f5d-82b25549ebd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna_folder = '../Data/RNA'\n",
    "dna_folder = '../Data/DNA'\n",
    "\n",
    "rna_files = os.listdir(rna_folder)\n",
    "dna_files = os.listdir(dna_folder)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3ce699f-a9fe-4557-8b50-4fb0279a675e",
   "metadata": {},
   "source": [
    "#### Map DNA to RNA file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "05bc8936-e798-4483-9b20-55f2c97b5e74",
   "metadata": {},
   "outputs": [],
   "source": [
    "terra_files_list = []\n",
    "terra_folder = '../Data/Terra_download_commands/'\n",
    "for file in os.listdir(terra_folder):\n",
    "    if file.endswith('.tsv'):\n",
    "        terra_df = pd.read_csv(os.path.join(terra_folder, file), sep = '\\t')\n",
    "        if file.startswith('LUAD'):\n",
    "            terra_df['DNA_file'] = terra_df['maf_file_capture_oxoG_filtered'].str.split('/').str[-1]\n",
    "        else:\n",
    "            terra_df['DNA_file'] = terra_df['maf_file_capture_novo_realign_filtered'].str.split('/').str[-1]\n",
    "        terra_df['RNA_file'] = terra_df['maf_file_rna_final_paper_v1_3'].str.split('/').str[-1]\n",
    "        terra_files_list.append(terra_df[['DNA_file', 'RNA_file']])\n",
    "\n",
    "terra_df = pd.concat(terra_files_list)\n",
    "terra_df = terra_df[((~terra_df['DNA_file'].isna()) & (~terra_df['RNA_file'].isna()))]\n",
    "\n",
    "# Initialize a dictionary where each value is a list\n",
    "dna_to_rna = defaultdict(list)\n",
    "\n",
    "# Populate the dictionary\n",
    "for dna, rna in zip(terra_df['DNA_file'], terra_df['RNA_file']):\n",
    "    dna_to_rna[dna].append(rna)\n",
    "\n",
    "# Convert defaultdict back to a regular dict if needed\n",
    "dna_to_rna = dict(dna_to_rna)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b63d5975-7f69-4c28-92e5-efd78eea25ea",
   "metadata": {},
   "source": [
    "#### Create a merged dataframe for DNA and RNA data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "24726634-3a60-4f6b-abb9-f2c7085da514",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (39,40,41,58) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:10: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  rna_sample_df = pd.read_csv(os.path.join(rna_folder, rna_file), sep = '\\t', encoding = 'latin1', usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:15: DtypeWarning: Columns (38,284,288,289,290,291) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1', skiprows=1, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:15: DtypeWarning: Columns (38,284,288,289,290,291) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1', skiprows=1, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:10: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  rna_sample_df = pd.read_csv(os.path.join(rna_folder, rna_file), sep = '\\t', encoding = 'latin1', usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:10: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  rna_sample_df = pd.read_csv(os.path.join(rna_folder, rna_file), sep = '\\t', encoding = 'latin1', usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:10: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  rna_sample_df = pd.read_csv(os.path.join(rna_folder, rna_file), sep = '\\t', encoding = 'latin1', usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:10: DtypeWarning: Columns (4,284,288,290,291) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  rna_sample_df = pd.read_csv(os.path.join(rna_folder, rna_file), sep = '\\t', encoding = 'latin1', usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:10: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  rna_sample_df = pd.read_csv(os.path.join(rna_folder, rna_file), sep = '\\t', encoding = 'latin1', usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:10: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  rna_sample_df = pd.read_csv(os.path.join(rna_folder, rna_file), sep = '\\t', encoding = 'latin1', usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\1814105258.py:12: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n"
     ]
    }
   ],
   "source": [
    "merged_df_list = []\n",
    "chosen_columns = ['Hugo_Symbol', 'Entrez_Gene_Id', 'Tumor_Sample_Barcode', 'Chromosome', 'Start_position', 'Variant_Classification', 'Reference_Allele', 'Tumor_Seq_Allele2', 'Transcript_Exon', 'Transcript_Position', 'cDNA_Change', 'Codon_Change', 'Protein_Change', 'COSMIC_tissue_types_affected', 'COSMIC_total_alterations_in_gene', 'ref_context', 'gc_content', 'i_COSMIC_n_overlapping_mutations', 'i_init_t_lod', 'i_t_lod_fstar','t_alt_count','t_ref_count','i_tumor_f']\n",
    "\n",
    "for dna_file in dna_to_rna:\n",
    "    for rna_file in dna_to_rna[dna_file]:\n",
    "        # if str(rna_file) == 'nan':\n",
    "        #     print(dna_file, \"doesn't exist in RNA data\")\n",
    "        #     continue\n",
    "        if rna_file.endswith('.txt'):\n",
    "            rna_sample_df = pd.read_csv(os.path.join(rna_folder, rna_file), sep = '\\t', encoding = 'latin1', usecols = chosen_columns)\n",
    "        if dna_file.endswith('.maf.annotated'):\n",
    "            dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1',  skiprows=3, usecols = chosen_columns)\n",
    "            dna_sample_df['i_t_lod_fstar'].astype(float)\n",
    "        elif dna_file.endswith('.maf'):\n",
    "            dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1', skiprows=1, usecols = chosen_columns)\n",
    "        elif dna_file.endswith('.txt'):\n",
    "            dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1', skiprows=1, usecols = chosen_columns)\n",
    "        # elif file.endswith('.txt'):\n",
    "        #     dna_sample_df = pd.read_csv(os.path.join(dna_folder, dna_file), sep = '\\t', encoding = 'latin1', skiprows=1, usecols = chosen_columns)\n",
    "\n",
    "        # dna_sample_df = dna_sample_df[chosen_columns]\n",
    "        # rna_sample_df = rna_sample_df[chosen_columns]\n",
    "        dna_sample_df.index = dna_sample_df['Chromosome'].astype(str) + '_' + dna_sample_df['Start_position'].astype(str) + '_' + dna_sample_df['Tumor_Seq_Allele2'].astype(str) \n",
    "        rna_sample_df.index = rna_sample_df['Chromosome'].astype(str) + '_' + rna_sample_df['Start_position'].astype(str) + '_' + rna_sample_df['Tumor_Seq_Allele2'].astype(str) \n",
    "        merged_sample_df = dna_sample_df.merge(rna_sample_df, left_index=True, right_index=True, how='outer', suffixes=['_DNA', '_RNA'])\n",
    "        merged_df_list.append(merged_sample_df)\n",
    "\n",
    "merged_df = pd.concat(merged_df_list)\n",
    "for col in ['i_init_t_lod_DNA','i_t_lod_fstar_DNA','t_alt_count_DNA', 't_ref_count_DNA', 'i_tumor_f_DNA']:\n",
    "    merged_df[col] = (merged_df[col].astype(str) + '|').str.split('|').str[0].astype(float)\n",
    "    merged_df = merged_df[merged_df['Tumor_Seq_Allele2_DNA'].str.len() == 1]\n",
    "merged_df.to_csv('../Output/DNA_RNA_merged.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bede043b-f4b9-478c-8aba-f91f44c9a2e6",
   "metadata": {},
   "source": [
    "### Format the data to create a simple predicting model for appearance of mutations in the RNA\n",
    "---\n",
    "General cancer expression data is calculated in Analyze_Expression_Data.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "da698cdf-760c-432c-b3f8-8ded15d1abe2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_23168\\4275542866.py:1: DtypeWarning: Columns (3,26) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  merged_df = pd.read_csv('../Output/DNA_RNA_merged.csv', index_col = 0)\n"
     ]
    }
   ],
   "source": [
    "merged_df = pd.read_csv('../Output/DNA_RNA_merged.csv', index_col = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e619b152-9b4e-4012-ad39-29a0d1e11bf1",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df['Entrez_Gene_Id_DNA'] = merged_df['Entrez_Gene_Id_DNA'].fillna(-1).astype(int).astype(str)\n",
    "merged_df = merged_df[~merged_df['Start_position_DNA'].isna()]\n",
    "merged_df['Appears_in_rna'] = (~merged_df['Start_position_RNA'].isna()).astype(int)\n",
    "merged_df = merged_df.iloc[:,~merged_df.columns.str.endswith('RNA')]\n",
    "merged_df['Cancer_type'] = merged_df['Tumor_Sample_Barcode_DNA'].str.split('-').str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "d990745d-3112-492c-901e-d3a46ac0a8fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('../Output/model_input.csv', index_col=0)\n",
    "expression_df = pd.read_csv('../Output/Expression.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "42ae08a2-6789-432b-a6d3-7cd12c9ea24b",
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_expression_by_cancer = expression_df.drop('Tumor_Sample_Barcode_DNA', axis = 1).groupby('Cancer_type').mean()\n",
    "\n",
    "# Step 2: Map mean expression values to the df dataframe\n",
    "# Assuming 'cancer_type' is a column in df as well\n",
    "# df['Mean_Expression'] = df['cancer_type'].map(mean_expression_by_cancer)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b0a787ca-b36d-4f8c-bb61-29a46193a175",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_df = pd.read_csv('../Output/Expression.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "56a9f6ff-28d7-4c68-ba0f-a967f97a36ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_type = expression_df['Cancer_type']\n",
    "sample = expression_df['Tumor_Sample_Barcode_DNA']\n",
    "expression_df = expression_df.drop(['Cancer_type', 'Tumor_Sample_Barcode_DNA'], axis = 1)\n",
    "expression_df.loc['Mean_Expression'] = expression_df.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "6672ef66-ee4e-4948-9f4c-3fbaa3bc1815",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TCGA-2F-A9KO-01A-11R-A38B-07    954.729786\n",
       "TCGA-2F-A9KP-01A-11R-A38B-07    910.201528\n",
       "TCGA-2F-A9KQ-01A-11R-A38B-07    914.055868\n",
       "TCGA-2F-A9KR-01A-11R-A38B-07    927.736279\n",
       "TCGA-2F-A9KT-01A-11R-A38B-07    945.576995\n",
       "                                   ...    \n",
       "TCGA-VQ-AA6I-01A-11R-A414-31    930.320116\n",
       "TCGA-VQ-AA6J-01A-11R-A414-31    888.309199\n",
       "TCGA-VQ-AA6K-01A-11R-A414-31    945.986212\n",
       "TCGA-ZA-A8F6-01A-23R-A36D-31    904.497547\n",
       "TCGA-ZQ-A9CR-01A-11R-A39E-31    951.395712\n",
       "Length: 2866, dtype: float64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_df.mean(axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "id": "817e2c9f-a3e6-4cf6-a2cb-a27911f85a96",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Cancer_type\n",
       "SKCM    402436\n",
       "LUAD    186978\n",
       "BLCA    136561\n",
       "STAD    134690\n",
       "COAD    119149\n",
       "HNSC     66509\n",
       "PAAD      7450\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 264,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['Cancer_type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 515,
   "id": "be2c74ea-ac5b-48e9-8a79-fb5da37cc6e3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.simplefilter(action='ignore', category=pd.errors.SettingWithCopyWarning)\n",
    "\n",
    "expression_data_folder = '../Data/Expression/'\n",
    "\n",
    "results_dict = {}\n",
    "files = os.listdir(expression_data_folder)\n",
    "files = [file for file in files if not file.startswith('LUAD')]\n",
    "for file in files:\n",
    "    if not file.endswith('.txt'):\n",
    "        continue\n",
    "    cancer_type = file.split('.')[0]\n",
    "    expression_df = pd.read_csv(os.path.join(expression_data_folder, file), sep = '\\t', index_col = 0, header = 0, skiprows = [1])\n",
    "    \n",
    "    cancer_type_df = merged_df[merged_df['Cancer_type'] == cancer_type]\n",
    "    expression_df.index = expression_df.index.str.split('|').str[1]\n",
    "    cancer_type_df['Shorten_barcode'] = cancer_type_df['Tumor_Sample_Barcode_DNA'].str.split('-').str[1:3].map('-'.join)\n",
    "    \n",
    "    shorten_barcode = expression_df.columns.str.split('-').str[1:3].map('-'.join)\n",
    "\n",
    "    kf = KFold(n_splits=5, shuffle=True, random_state=42)\n",
    "\n",
    "    samples = np.array(cancer_type_df['Shorten_barcode'].unique())\n",
    "    \n",
    "    fold_results = []\n",
    "    \n",
    "    for train_index, test_index in kf.split(samples):\n",
    "        temp_expression_df = expression_df.loc[:,~shorten_barcode.isin(samples[train_index])]\n",
    "    \n",
    "        # temp_expression_df['Mean_Expression'] = temp_expression_df.mean(axis = 1)\n",
    "        # temp_cancer_type_df = cancer_type_df.merge(temp_expression_df[['Mean_Expression']], left_on = 'Entrez_Gene_Id_DNA', right_index=True)\n",
    "        \n",
    "        train_cancer_type_df = temp_cancer_type_df.loc[temp_cancer_type_df['Shorten_barcode'].isin(samples[train_index]),:]\n",
    "        test_cancer_type_df = temp_cancer_type_df.loc[temp_cancer_type_df['Shorten_barcode'].isin(samples[test_index]),:]\n",
    "        \n",
    "        X_train, y_train = train_cancer_type_df[['Mean_Expression']].values, train_cancer_type_df['Appears_in_rna']\n",
    "        X_test, y_test = test_cancer_type_df[['Mean_Expression']].values, test_cancer_type_df['Appears_in_rna']\n",
    "        model = LogisticRegression()\n",
    "        # Train the model\n",
    "        model.fit(X_train, y_train)\n",
    "        \n",
    "        # Make predictions\n",
    "        y_pred = model.predict(X_test)\n",
    "        y_pred_prob = model.predict_proba(X_test)[:, 1]\n",
    "        \n",
    "        # Evaluate the model\n",
    "        accuracy = accuracy_score(y_test, y_pred)\n",
    "        roc_auc = roc_auc_score(y_test, y_pred_prob)\n",
    "        \n",
    "        fold_results.append({'accuracy': accuracy, 'roc_auc': roc_auc})\n",
    "        \n",
    "    # Compile results\n",
    "    results_df = pd.DataFrame(fold_results)\n",
    "    results_dict[cancer_type] = results_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d664d10-ac60-489f-a025-e493b1800220",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Variant types that do not exist in the RNA:  'Intron', RNA?, 'IGR', lincRNA?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 516,
   "id": "0fb193c6-d08c-45af-ae6b-37132aadcbf6",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results for cancer BLCA:\n",
      "   accuracy   roc_auc\n",
      "0  0.689010  0.806451\n",
      "1  0.713347  0.802415\n",
      "2  0.693296  0.808705\n",
      "3  0.684852  0.798996\n",
      "4  0.685488  0.797563\n",
      "Mean accuracy:  0.6931986579422642 , Mean roc_auc:  0.8028258907794351\n",
      "\n",
      "Results for cancer COAD:\n",
      "   accuracy   roc_auc\n",
      "0  0.714474  0.861973\n",
      "1  0.729282  0.860628\n",
      "2  0.708526  0.871761\n",
      "3  0.709386  0.859770\n",
      "4  0.716886  0.846626\n",
      "Mean accuracy:  0.7157108081919978 , Mean roc_auc:  0.8601517115491382\n",
      "\n",
      "Results for cancer HNSC:\n",
      "   accuracy   roc_auc\n",
      "0  0.685306  0.829730\n",
      "1  0.712544  0.833500\n",
      "2  0.704988  0.834233\n",
      "3  0.700223  0.810802\n",
      "4  0.980809  0.681440\n",
      "Mean accuracy:  0.7567740126632092 , Mean roc_auc:  0.7979410689139673\n",
      "\n",
      "Results for cancer PAAD:\n",
      "   accuracy   roc_auc\n",
      "0  0.786458  0.817521\n",
      "1  0.738725  0.821986\n",
      "2  0.756003  0.805413\n",
      "3  0.781272  0.814206\n",
      "4  0.780285  0.814211\n",
      "Mean accuracy:  0.7685486066100021 , Mean roc_auc:  0.8146674373294065\n",
      "\n",
      "Results for cancer SKCM:\n",
      "   accuracy   roc_auc\n",
      "0  0.799533  0.866210\n",
      "1  0.787107  0.864907\n",
      "2  0.795400  0.870543\n",
      "3  0.776307  0.863400\n",
      "4  0.791053  0.861691\n",
      "Mean accuracy:  0.7898799647668937 , Mean roc_auc:  0.8653501849692112\n",
      "\n",
      "Results for cancer STAD:\n",
      "   accuracy   roc_auc\n",
      "0  0.645918  0.817730\n",
      "1  0.654579  0.820758\n",
      "2  0.651851  0.828168\n",
      "3  0.643878  0.821315\n",
      "4  0.637408  0.827727\n",
      "Mean accuracy:  0.6467266703406015 , Mean roc_auc:  0.8231396806360589\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for key,value in results_dict.items():\n",
    "    print(f'Results for cancer {key}:')\n",
    "    print(value)\n",
    "    print('Mean accuracy: ', value['accuracy'].mean(), ', Mean roc_auc: ', value['roc_auc'].mean())\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "44dc1a49-1231-4413-a286-161062933abe",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_df = pd.read_csv('../Output/mean_expression_per_cancer.csv', index_col = 0)\n",
    "expression_df.index = expression_df.index.str.split('|').str[1]\n",
    "expression_df = pd.DataFrame({\n",
    "    'Entrez_Gene_Id': expression_df.index.repeat(len(expression_df.columns)),\n",
    "    'Cancer_type': list(expression_df.columns) * len(expression_df),\n",
    "    'Expression': expression_df.values.flatten()\n",
    "})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "d0e7eadd-a9f9-4290-8b1b-ec0caf0bc690",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = expression_df.merge(X[['Entrez_Gene_Id_DNA', 'Cancer_type', 'Appears_in_rna', 'Tumor_Sample_Barcode_DNA']], \n",
    "                    left_on = ['Entrez_Gene_Id', 'Cancer_type'],\n",
    "                   right_on = ['Entrez_Gene_Id_DNA', 'Cancer_type'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "84db463f-2827-45cd-960e-11726252d74c",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cross-validation results for COAD:\n",
      "   accuracy   roc_auc\n",
      "0  0.717672  0.862840\n",
      "1  0.717020  0.854323\n",
      "2  0.717455  0.858084\n",
      "3  0.714106  0.861688\n",
      "4  0.723283  0.857485\n",
      "\n",
      "Mean Accuracy: 0.7179070070897308\n",
      "Mean ROC AUC: 0.8588843304093918\n",
      "\n",
      "Cross-validation results for SKCM:\n",
      "   accuracy   roc_auc\n",
      "0  0.789754  0.866709\n",
      "1  0.789909  0.866788\n",
      "2  0.792509  0.867628\n",
      "3  0.789699  0.863847\n",
      "4  0.789104  0.866356\n",
      "\n",
      "Mean Accuracy: 0.7901949124859613\n",
      "Mean ROC AUC: 0.8662655544415555\n",
      "\n",
      "Cross-validation results for STAD:\n",
      "   accuracy   roc_auc\n",
      "0  0.644788  0.820764\n",
      "1  0.648721  0.825097\n",
      "2  0.646697  0.822936\n",
      "3  0.645743  0.821055\n",
      "4  0.647843  0.821634\n",
      "\n",
      "Mean Accuracy: 0.646758304696449\n",
      "Mean ROC AUC: 0.8222970982316411\n",
      "\n",
      "Cross-validation results for BLCA:\n",
      "   accuracy   roc_auc\n",
      "0  0.695116  0.801937\n",
      "1  0.691725  0.805913\n",
      "2  0.690067  0.803845\n",
      "3  0.694201  0.802267\n",
      "4  0.694766  0.797857\n",
      "\n",
      "Mean Accuracy: 0.6931749865577578\n",
      "Mean ROC AUC: 0.8023638046327303\n",
      "\n",
      "Cross-validation results for HNSC:\n",
      "   accuracy   roc_auc\n",
      "0  0.937676  0.697289\n",
      "1  0.940492  0.695498\n",
      "2  0.940004  0.696955\n",
      "3  0.940507  0.697422\n",
      "4  0.938777  0.697892\n",
      "\n",
      "Mean Accuracy: 0.9394912774692863\n",
      "Mean ROC AUC: 0.6970112157710566\n",
      "\n",
      "Cross-validation results for PAAD:\n",
      "   accuracy   roc_auc\n",
      "0  0.778409  0.822884\n",
      "1  0.768301  0.811416\n",
      "2  0.769012  0.815118\n",
      "3  0.767591  0.812573\n",
      "4  0.759773  0.809022\n",
      "\n",
      "Mean Accuracy: 0.7686171415649026\n",
      "Mean ROC AUC: 0.81420278802659\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Simple cross validation by mutations (and not by sample)\n",
    "import pandas as pd\n",
    "from sklearn.model_selection import KFold\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.metrics import accuracy_score, roc_auc_score, confusion_matrix\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "\n",
    "for cancer_type in df['Cancer_type'].unique():\n",
    "    cancer_type_df = df[df['Cancer_type'] == cancer_type]\n",
    "    # Define features (X) and target (y)\n",
    "    X = cancer_type_df[['Expression']].values\n",
    "    y = cancer_type_df['Appears_in_rna'].values\n",
    "    \n",
    "    # Standardize the feature\n",
    "    scaler = StandardScaler()\n",
    "    X = scaler.fit_transform(X)\n",
    "    \n",
    "    # Set up five-fold cross-validation\n",
    "    kf = KFold(n_splits=5, shuffle=True, random_state=42)\n",
    "    \n",
    "    # Initialize the model\n",
    "    model = LogisticRegression()\n",
    "    \n",
    "    # Cross-validation loop\n",
    "    fold_results = []\n",
    "    for train_index, test_index in kf.split(X):\n",
    "        # Split the data into training and testing sets\n",
    "        X_train, X_test = X[train_index], X[test_index]\n",
    "        y_train, y_test = y[train_index], y[test_index]\n",
    "        \n",
    "        # Train the model\n",
    "        model.fit(X_train, y_train)\n",
    "        \n",
    "        # Make predictions\n",
    "        y_pred = model.predict(X_test)\n",
    "        y_pred_prob = model.predict_proba(X_test)[:, 1]\n",
    "        \n",
    "        # Evaluate the model\n",
    "        accuracy = accuracy_score(y_test, y_pred)\n",
    "        roc_auc = roc_auc_score(y_test, y_pred_prob)\n",
    "        # Generate the confusion matrix\n",
    "        cm = confusion_matrix(y_test, y_pred)\n",
    "        \n",
    "        fold_results.append(({'accuracy': accuracy, 'roc_auc': roc_auc}, cm))\n",
    "    \n",
    "    # Compile results\n",
    "    results_df = pd.DataFrame(fold_results[fold[0] for fold in fold_results])\n",
    "    \n",
    "    print(f\"Cross-validation results for {cancer_type}:\")\n",
    "    print(results_df)\n",
    "    print(\"\\nMean Accuracy:\", results_df['accuracy'].mean())\n",
    "    print(\"Mean ROC AUC:\", results_df['roc_auc'].mean())\n",
    "    print()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e156b4d9-11af-41aa-8483-0b7bd79307d1",
   "metadata": {},
   "source": [
    "#### Format data for full model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6f2b5f29-f702-4a32-b50d-a69cc7e9a713",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\KerenYlab.MEDICINE\\AppData\\Local\\Temp\\ipykernel_4120\\4275542866.py:1: DtypeWarning: Columns (3,26) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  merged_df = pd.read_csv('../Output/DNA_RNA_merged.csv', index_col = 0)\n"
     ]
    }
   ],
   "source": [
    "merged_df = pd.read_csv('../Output/DNA_RNA_merged.csv', index_col = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2601dcf1-8f48-4148-837c-713c90082656",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df['Entrez_Gene_Id_DNA'] = merged_df['Entrez_Gene_Id_DNA'].fillna(-1).astype(int).astype(str)\n",
    "merged_df = merged_df[~merged_df['Start_position_DNA'].isna()]\n",
    "merged_df['Appears_in_rna'] = (~merged_df['Start_position_RNA'].isna()).astype(int)\n",
    "merged_df = merged_df.iloc[:,~merged_df.columns.str.endswith('RNA')]\n",
    "merged_df['Cancer_type'] = merged_df['Tumor_Sample_Barcode_DNA'].str.split('-').str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 579,
   "id": "870823a5-bbe5-4843-a4ff-4dbb08617444",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol_DNA</th>\n",
       "      <th>Entrez_Gene_Id_DNA</th>\n",
       "      <th>Chromosome_DNA</th>\n",
       "      <th>Start_position_DNA</th>\n",
       "      <th>Variant_Classification_DNA</th>\n",
       "      <th>Reference_Allele_DNA</th>\n",
       "      <th>Tumor_Seq_Allele2_DNA</th>\n",
       "      <th>Tumor_Sample_Barcode_DNA</th>\n",
       "      <th>Transcript_Exon_DNA</th>\n",
       "      <th>Transcript_Position_DNA</th>\n",
       "      <th>...</th>\n",
       "      <th>ref_context_DNA</th>\n",
       "      <th>gc_content_DNA</th>\n",
       "      <th>i_COSMIC_n_overlapping_mutations_DNA</th>\n",
       "      <th>i_init_t_lod_DNA</th>\n",
       "      <th>i_t_lod_fstar_DNA</th>\n",
       "      <th>t_alt_count_DNA</th>\n",
       "      <th>t_ref_count_DNA</th>\n",
       "      <th>i_tumor_f_DNA</th>\n",
       "      <th>Appears_in_rna</th>\n",
       "      <th>Cancer_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10_101715548_T</th>\n",
       "      <td>DNMBP</td>\n",
       "      <td>23268</td>\n",
       "      <td>10</td>\n",
       "      <td>101715548.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1774.0</td>\n",
       "      <td>...</td>\n",
       "      <td>CTGCCAAGCTCTTCTCAAACT</td>\n",
       "      <td>0.493</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.074207</td>\n",
       "      <td>40.720366</td>\n",
       "      <td>18.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>0.170213</td>\n",
       "      <td>0</td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10_102738692_C</th>\n",
       "      <td>SEMA4G</td>\n",
       "      <td>57715</td>\n",
       "      <td>10</td>\n",
       "      <td>102738692.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1103.0</td>\n",
       "      <td>...</td>\n",
       "      <td>CTTCTTCACGGAGCGTGCCAC</td>\n",
       "      <td>0.602</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.107030</td>\n",
       "      <td>6.309237</td>\n",
       "      <td>3.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>0.075000</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10_102822569_A</th>\n",
       "      <td>KAZALD1</td>\n",
       "      <td>81621</td>\n",
       "      <td>10</td>\n",
       "      <td>102822569.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>2.0</td>\n",
       "      <td>546.0</td>\n",
       "      <td>...</td>\n",
       "      <td>CAGGGTGCGCGACGCGTGCGG</td>\n",
       "      <td>0.736</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.275782</td>\n",
       "      <td>9.656761</td>\n",
       "      <td>4.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.111111</td>\n",
       "      <td>0</td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10_103826020_T</th>\n",
       "      <td>HPS6</td>\n",
       "      <td>79803</td>\n",
       "      <td>10</td>\n",
       "      <td>103826020.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>874.0</td>\n",
       "      <td>...</td>\n",
       "      <td>GGTTGCTGTCCCCCAGGGAGC</td>\n",
       "      <td>0.632</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.478583</td>\n",
       "      <td>16.651170</td>\n",
       "      <td>8.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.112903</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10_104160055_C</th>\n",
       "      <td>NFKB2</td>\n",
       "      <td>4791</td>\n",
       "      <td>10</td>\n",
       "      <td>104160055.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1855.0</td>\n",
       "      <td>...</td>\n",
       "      <td>ACCTGGCGGTGATCACGGGGC</td>\n",
       "      <td>0.662</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.040048</td>\n",
       "      <td>36.305156</td>\n",
       "      <td>14.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.188406</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8_77763180_T</th>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>79776</td>\n",
       "      <td>8</td>\n",
       "      <td>77763180.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4471.0</td>\n",
       "      <td>...</td>\n",
       "      <td>AGGCTAATGTGGAAGTAAAGA</td>\n",
       "      <td>0.403</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.208521</td>\n",
       "      <td>7.264761</td>\n",
       "      <td>3.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.130435</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9_100881456_G</th>\n",
       "      <td>TRIM14</td>\n",
       "      <td>9830</td>\n",
       "      <td>9</td>\n",
       "      <td>100881456.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>...</td>\n",
       "      <td>GGCTCCCGGTCGCCGCGCCCG</td>\n",
       "      <td>0.731</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.013856</td>\n",
       "      <td>6.362598</td>\n",
       "      <td>3.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>1</td>\n",
       "      <td>STAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9_32631933_A</th>\n",
       "      <td>TAF1L</td>\n",
       "      <td>138474</td>\n",
       "      <td>9</td>\n",
       "      <td>32631933.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3734.0</td>\n",
       "      <td>...</td>\n",
       "      <td>TAGTCCGTATGCGCACATAGG</td>\n",
       "      <td>0.433</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43.645163</td>\n",
       "      <td>60.755127</td>\n",
       "      <td>29.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>0.163522</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_105179181_C</th>\n",
       "      <td>NRK</td>\n",
       "      <td>203447</td>\n",
       "      <td>X</td>\n",
       "      <td>105179181.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>21.0</td>\n",
       "      <td>3822.0</td>\n",
       "      <td>...</td>\n",
       "      <td>GATTCGTAGAAGTACCTGAGG</td>\n",
       "      <td>0.378</td>\n",
       "      <td>0.0</td>\n",
       "      <td>69.450299</td>\n",
       "      <td>74.950507</td>\n",
       "      <td>30.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>0.261682</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_135496331_A</th>\n",
       "      <td>GPR112</td>\n",
       "      <td>139378</td>\n",
       "      <td>X</td>\n",
       "      <td>135496331.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>25.0</td>\n",
       "      <td>9341.0</td>\n",
       "      <td>...</td>\n",
       "      <td>GGGAGCAGCCGGTGTCAGATA</td>\n",
       "      <td>0.353</td>\n",
       "      <td>0.0</td>\n",
       "      <td>105.449508</td>\n",
       "      <td>108.299788</td>\n",
       "      <td>39.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>0.330357</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1127195 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Hugo_Symbol_DNA Entrez_Gene_Id_DNA Chromosome_DNA  \\\n",
       "10_101715548_T           DNMBP              23268             10   \n",
       "10_102738692_C          SEMA4G              57715             10   \n",
       "10_102822569_A         KAZALD1              81621             10   \n",
       "10_103826020_T            HPS6              79803             10   \n",
       "10_104160055_C           NFKB2               4791             10   \n",
       "...                        ...                ...            ...   \n",
       "8_77763180_T             ZFHX4              79776              8   \n",
       "9_100881456_G           TRIM14               9830              9   \n",
       "9_32631933_A             TAF1L             138474              9   \n",
       "X_105179181_C              NRK             203447              X   \n",
       "X_135496331_A           GPR112             139378              X   \n",
       "\n",
       "                Start_position_DNA Variant_Classification_DNA  \\\n",
       "10_101715548_T         101715548.0                     Silent   \n",
       "10_102738692_C         102738692.0          Missense_Mutation   \n",
       "10_102822569_A         102822569.0          Missense_Mutation   \n",
       "10_103826020_T         103826020.0                     Silent   \n",
       "10_104160055_C         104160055.0                     Silent   \n",
       "...                            ...                        ...   \n",
       "8_77763180_T            77763180.0                     Silent   \n",
       "9_100881456_G          100881456.0                     Silent   \n",
       "9_32631933_A            32631933.0                     Silent   \n",
       "X_105179181_C          105179181.0          Missense_Mutation   \n",
       "X_135496331_A          135496331.0          Missense_Mutation   \n",
       "\n",
       "               Reference_Allele_DNA Tumor_Seq_Allele2_DNA  \\\n",
       "10_101715548_T                    C                     T   \n",
       "10_102738692_C                    G                     C   \n",
       "10_102822569_A                    G                     A   \n",
       "10_103826020_T                    C                     T   \n",
       "10_104160055_C                    G                     C   \n",
       "...                             ...                   ...   \n",
       "8_77763180_T                      G                     T   \n",
       "9_100881456_G                     C                     G   \n",
       "9_32631933_A                      G                     A   \n",
       "X_105179181_C                     A                     C   \n",
       "X_135496331_A                     G                     A   \n",
       "\n",
       "               Tumor_Sample_Barcode_DNA  Transcript_Exon_DNA  \\\n",
       "10_101715548_T          BLCA-2F-A9KO-TP                  4.0   \n",
       "10_102738692_C          BLCA-2F-A9KO-TP                  7.0   \n",
       "10_102822569_A          BLCA-2F-A9KO-TP                  2.0   \n",
       "10_103826020_T          BLCA-2F-A9KO-TP                  1.0   \n",
       "10_104160055_C          BLCA-2F-A9KO-TP                 16.0   \n",
       "...                                 ...                  ...   \n",
       "8_77763180_T            STAD-ZQ-A9CR-TP                 10.0   \n",
       "9_100881456_G           STAD-ZQ-A9CR-TP                  1.0   \n",
       "9_32631933_A            STAD-ZQ-A9CR-TP                  1.0   \n",
       "X_105179181_C           STAD-ZQ-A9CR-TP                 21.0   \n",
       "X_135496331_A           STAD-ZQ-A9CR-TP                 25.0   \n",
       "\n",
       "                Transcript_Position_DNA  ...        ref_context_DNA  \\\n",
       "10_101715548_T                   1774.0  ...  CTGCCAAGCTCTTCTCAAACT   \n",
       "10_102738692_C                   1103.0  ...  CTTCTTCACGGAGCGTGCCAC   \n",
       "10_102822569_A                    546.0  ...  CAGGGTGCGCGACGCGTGCGG   \n",
       "10_103826020_T                    874.0  ...  GGTTGCTGTCCCCCAGGGAGC   \n",
       "10_104160055_C                   1855.0  ...  ACCTGGCGGTGATCACGGGGC   \n",
       "...                                 ...  ...                    ...   \n",
       "8_77763180_T                     4471.0  ...  AGGCTAATGTGGAAGTAAAGA   \n",
       "9_100881456_G                      24.0  ...  GGCTCCCGGTCGCCGCGCCCG   \n",
       "9_32631933_A                     3734.0  ...  TAGTCCGTATGCGCACATAGG   \n",
       "X_105179181_C                    3822.0  ...  GATTCGTAGAAGTACCTGAGG   \n",
       "X_135496331_A                    9341.0  ...  GGGAGCAGCCGGTGTCAGATA   \n",
       "\n",
       "               gc_content_DNA i_COSMIC_n_overlapping_mutations_DNA  \\\n",
       "10_101715548_T          0.493                                  0.0   \n",
       "10_102738692_C          0.602                                  0.0   \n",
       "10_102822569_A          0.736                                  0.0   \n",
       "10_103826020_T          0.632                                  0.0   \n",
       "10_104160055_C          0.662                                  0.0   \n",
       "...                       ...                                  ...   \n",
       "8_77763180_T            0.403                                  0.0   \n",
       "9_100881456_G           0.731                                  0.0   \n",
       "9_32631933_A            0.433                                  0.0   \n",
       "X_105179181_C           0.378                                  0.0   \n",
       "X_135496331_A           0.353                                  0.0   \n",
       "\n",
       "               i_init_t_lod_DNA  i_t_lod_fstar_DNA t_alt_count_DNA  \\\n",
       "10_101715548_T        31.074207          40.720366            18.0   \n",
       "10_102738692_C        -1.107030           6.309237             3.0   \n",
       "10_102822569_A         4.275782           9.656761             4.0   \n",
       "10_103826020_T         7.478583          16.651170             8.0   \n",
       "10_104160055_C        30.040048          36.305156            14.0   \n",
       "...                         ...                ...             ...   \n",
       "8_77763180_T           4.208521           7.264761             3.0   \n",
       "9_100881456_G          4.013856           6.362598             3.0   \n",
       "9_32631933_A          43.645163          60.755127            29.0   \n",
       "X_105179181_C         69.450299          74.950507            30.0   \n",
       "X_135496331_A        105.449508         108.299788            39.0   \n",
       "\n",
       "                t_ref_count_DNA  i_tumor_f_DNA  Appears_in_rna  Cancer_type  \n",
       "10_101715548_T             78.0       0.170213               0         BLCA  \n",
       "10_102738692_C             37.0       0.075000               1         BLCA  \n",
       "10_102822569_A             33.0       0.111111               0         BLCA  \n",
       "10_103826020_T             56.0       0.112903               1         BLCA  \n",
       "10_104160055_C             56.0       0.188406               1         BLCA  \n",
       "...                         ...            ...             ...          ...  \n",
       "8_77763180_T               21.0       0.130435               0         STAD  \n",
       "9_100881456_G              17.0       0.150000               1         STAD  \n",
       "9_32631933_A              137.0       0.163522               0         STAD  \n",
       "X_105179181_C              80.0       0.261682               0         STAD  \n",
       "X_135496331_A              76.0       0.330357               0         STAD  \n",
       "\n",
       "[1127195 rows x 25 columns]"
      ]
     },
     "execution_count": 579,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Non-features:\n",
    "Tumor_Sample_Barcode_DNA\n",
    "\n",
    "Features:\n",
    "'Gene' - need to think of an embedding. Gene2Vec?\n",
    "'Chromosome_DNA' - One-hot - remove some annoying mutations and make chromosomes strings\n",
    "'Start_position_DNA' - Float - scaling\n",
    "'Variant_Classification_DNA' - One-hot\n",
    "'Reference_Allele_DNA' - One-hot - Maybe collapse by strand?\n",
    "'Tumor_Seq_Allele2_DNA' - One-hot - Maybe collapse by strand? - Maybe remove mutations that are different that 'ref_context_DNA'  # merged_df[(merged_df['Reference_Allele_DNA'] != merged_df['ref_context_DNA'].str.slice(10,11).str.upper())]\n",
    "'Transcript_Exon_DNA' - In logistic regression maybe turn to binning and one-hot\n",
    "'Transcript_Position_DNA' - Float - scaling\n",
    "'Gene Expression'\t- Float\n",
    "\n",
    "'Protein_Change_DNA' - Protein substitution, Myabe one hot, maybe reading frame instead\n",
    "'COSMIC_tissue_types_affected_DNA' - One-hot, on all tissues\n",
    "'COSMIC_total_alterations_in_gene_DNA' - Float\n",
    "\n",
    "'ref_context_DNA' - One-hot of the left and right bases\n",
    "'gc_content_DNA' - Float\n",
    "'i_COSMIC_n_overlapping_mutations_DNA' - Float\n",
    "'i_init_t_lod_DNA'- Float\n",
    "'i_t_lod_fstar_DNA'- Float\n",
    "'t_alt_count_DNA'- Float\n",
    "'t_ref_count_DNA'- Float\n",
    "'i_tumor_f_DNA'\t- Float\n",
    "'Cancer_type' - One-hot"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92884a2a-2841-462c-995a-037cd7207567",
   "metadata": {},
   "source": [
    "#### Filter out the standard chromosomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "de2c76c7-a6c7-47e8-841a-523cb30eead1",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df['Chromosome_DNA'] = merged_df['Chromosome_DNA'].astype(str).str.replace('.0','')\n",
    "chromosomes = set([str(i) for i in range(1,23)] + ['X', 'Y'])\n",
    "merged_df = merged_df[merged_df['Chromosome_DNA'].isin(chromosomes)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50a08767-e8b0-4737-8cf3-4961b0631383",
   "metadata": {},
   "source": [
    "#### Filter out by the variant type - need to make sure this is OK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1f7c8ca4-a2a8-4b09-8dcb-13a9f9aeacaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "variant_removal = merged_df[['Variant_Classification_DNA', 'Appears_in_rna']].groupby('Variant_Classification_DNA').mean()\n",
    "non_zero_relevant = list(variant_removal[variant_removal['Appears_in_rna'] != 0].index)\n",
    "merged_df = merged_df[merged_df['Variant_Classification_DNA'].isin(non_zero_relevant)] # remove RNA, and lincRNA?\n",
    "merged_df['Left_flank_base'] =  merged_df['ref_context_DNA'].str.slice(9,10).str.upper()\n",
    "merged_df['Right_flank_base'] =  merged_df['ref_context_DNA'].str.slice(11,12).str.upper()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "235c3814-7b87-48cd-9811-a7438c6096b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "tissues_dict = {}\n",
    "for index, row in merged_df['COSMIC_tissue_types_affected_DNA'][~merged_df['COSMIC_tissue_types_affected_DNA'].isna()].str.split('|').items():\n",
    "    for tissues in row:\n",
    "        tissue = tissues[:tissues.index('(')]\n",
    "        number = tissues[tissues.index('(')+1:tissues.index(')')]\n",
    "        if tissue not in tissues_dict:\n",
    "            tissues_dict[tissue] = {}\n",
    "        try:\n",
    "            number = int(number)\n",
    "        except:\n",
    "            number = 0\n",
    "        tissues_dict[tissue][index] = number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "50ba5d49-0938-411c-93bd-6964d4f8094e",
   "metadata": {},
   "outputs": [],
   "source": [
    "existing_samples = merged_df['Tumor_Sample_Barcode_DNA'].str.split('-').str[1:3].map('-'.join).unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "cf148530-5cd3-421a-9403-e703a5f9f0a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_df = pd.read_csv('../Output/mean_expression_per_cancer.csv', index_col = 0)\n",
    "expression_df.index = expression_df.index.str.split('|').str[1]\n",
    "expression_df = pd.DataFrame({\n",
    "    'Entrez_Gene_Id': expression_df.index.repeat(len(expression_df.columns)),\n",
    "    'Cancer_type': list(expression_df.columns) * len(expression_df),\n",
    "    'Expression': expression_df.values.flatten()\n",
    "})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "dfbd3874-b61e-471e-aa66-a10a1baf07b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol_DNA</th>\n",
       "      <th>Entrez_Gene_Id_DNA</th>\n",
       "      <th>Chromosome_DNA</th>\n",
       "      <th>Start_position_DNA</th>\n",
       "      <th>Variant_Classification_DNA</th>\n",
       "      <th>Reference_Allele_DNA</th>\n",
       "      <th>Tumor_Seq_Allele2_DNA</th>\n",
       "      <th>Tumor_Sample_Barcode_DNA</th>\n",
       "      <th>Transcript_Exon_DNA</th>\n",
       "      <th>Transcript_Position_DNA</th>\n",
       "      <th>...</th>\n",
       "      <th>i_t_lod_fstar_DNA</th>\n",
       "      <th>t_alt_count_DNA</th>\n",
       "      <th>t_ref_count_DNA</th>\n",
       "      <th>i_tumor_f_DNA</th>\n",
       "      <th>Appears_in_rna</th>\n",
       "      <th>Cancer_type</th>\n",
       "      <th>Left_flank_base</th>\n",
       "      <th>Right_flank_base</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Expression</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DNMBP</td>\n",
       "      <td>23268</td>\n",
       "      <td>10</td>\n",
       "      <td>101715548.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1774.0</td>\n",
       "      <td>...</td>\n",
       "      <td>40.720366</td>\n",
       "      <td>18.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>0.170213</td>\n",
       "      <td>0</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>23268</td>\n",
       "      <td>893.592239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SEMA4G</td>\n",
       "      <td>57715</td>\n",
       "      <td>10</td>\n",
       "      <td>102738692.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1103.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.309237</td>\n",
       "      <td>3.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>0.075000</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>57715</td>\n",
       "      <td>162.400803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KAZALD1</td>\n",
       "      <td>81621</td>\n",
       "      <td>10</td>\n",
       "      <td>102822569.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>2.0</td>\n",
       "      <td>546.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9.656761</td>\n",
       "      <td>4.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.111111</td>\n",
       "      <td>0</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>81621</td>\n",
       "      <td>61.660977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HPS6</td>\n",
       "      <td>79803</td>\n",
       "      <td>10</td>\n",
       "      <td>103826020.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>874.0</td>\n",
       "      <td>...</td>\n",
       "      <td>16.651170</td>\n",
       "      <td>8.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.112903</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>79803</td>\n",
       "      <td>566.412410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NFKB2</td>\n",
       "      <td>4791</td>\n",
       "      <td>10</td>\n",
       "      <td>104160055.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1855.0</td>\n",
       "      <td>...</td>\n",
       "      <td>36.305156</td>\n",
       "      <td>14.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.188406</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>4791</td>\n",
       "      <td>1578.509691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968891</th>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>79776</td>\n",
       "      <td>8</td>\n",
       "      <td>77763180.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4471.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.264761</td>\n",
       "      <td>3.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.130435</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>79776</td>\n",
       "      <td>255.884184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968892</th>\n",
       "      <td>TRIM14</td>\n",
       "      <td>9830</td>\n",
       "      <td>9</td>\n",
       "      <td>100881456.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.362598</td>\n",
       "      <td>3.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>1</td>\n",
       "      <td>STAD</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>9830</td>\n",
       "      <td>1640.420685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968893</th>\n",
       "      <td>TAF1L</td>\n",
       "      <td>138474</td>\n",
       "      <td>9</td>\n",
       "      <td>32631933.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3734.0</td>\n",
       "      <td>...</td>\n",
       "      <td>60.755127</td>\n",
       "      <td>29.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>0.163522</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>138474</td>\n",
       "      <td>40.830964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968894</th>\n",
       "      <td>NRK</td>\n",
       "      <td>203447</td>\n",
       "      <td>X</td>\n",
       "      <td>105179181.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>21.0</td>\n",
       "      <td>3822.0</td>\n",
       "      <td>...</td>\n",
       "      <td>74.950507</td>\n",
       "      <td>30.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>0.261682</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>203447</td>\n",
       "      <td>82.048767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968895</th>\n",
       "      <td>GPR112</td>\n",
       "      <td>139378</td>\n",
       "      <td>X</td>\n",
       "      <td>135496331.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>25.0</td>\n",
       "      <td>9341.0</td>\n",
       "      <td>...</td>\n",
       "      <td>108.299788</td>\n",
       "      <td>39.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>0.330357</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>139378</td>\n",
       "      <td>6.223744</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>968896 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol_DNA Entrez_Gene_Id_DNA Chromosome_DNA  Start_position_DNA  \\\n",
       "0                DNMBP              23268             10         101715548.0   \n",
       "1               SEMA4G              57715             10         102738692.0   \n",
       "2              KAZALD1              81621             10         102822569.0   \n",
       "3                 HPS6              79803             10         103826020.0   \n",
       "4                NFKB2               4791             10         104160055.0   \n",
       "...                ...                ...            ...                 ...   \n",
       "968891           ZFHX4              79776              8          77763180.0   \n",
       "968892          TRIM14               9830              9         100881456.0   \n",
       "968893           TAF1L             138474              9          32631933.0   \n",
       "968894             NRK             203447              X         105179181.0   \n",
       "968895          GPR112             139378              X         135496331.0   \n",
       "\n",
       "       Variant_Classification_DNA Reference_Allele_DNA Tumor_Seq_Allele2_DNA  \\\n",
       "0                          Silent                    C                     T   \n",
       "1               Missense_Mutation                    G                     C   \n",
       "2               Missense_Mutation                    G                     A   \n",
       "3                          Silent                    C                     T   \n",
       "4                          Silent                    G                     C   \n",
       "...                           ...                  ...                   ...   \n",
       "968891                     Silent                    G                     T   \n",
       "968892                     Silent                    C                     G   \n",
       "968893                     Silent                    G                     A   \n",
       "968894          Missense_Mutation                    A                     C   \n",
       "968895          Missense_Mutation                    G                     A   \n",
       "\n",
       "       Tumor_Sample_Barcode_DNA  Transcript_Exon_DNA Transcript_Position_DNA  \\\n",
       "0               BLCA-2F-A9KO-TP                  4.0                  1774.0   \n",
       "1               BLCA-2F-A9KO-TP                  7.0                  1103.0   \n",
       "2               BLCA-2F-A9KO-TP                  2.0                   546.0   \n",
       "3               BLCA-2F-A9KO-TP                  1.0                   874.0   \n",
       "4               BLCA-2F-A9KO-TP                 16.0                  1855.0   \n",
       "...                         ...                  ...                     ...   \n",
       "968891          STAD-ZQ-A9CR-TP                 10.0                  4471.0   \n",
       "968892          STAD-ZQ-A9CR-TP                  1.0                    24.0   \n",
       "968893          STAD-ZQ-A9CR-TP                  1.0                  3734.0   \n",
       "968894          STAD-ZQ-A9CR-TP                 21.0                  3822.0   \n",
       "968895          STAD-ZQ-A9CR-TP                 25.0                  9341.0   \n",
       "\n",
       "        ... i_t_lod_fstar_DNA t_alt_count_DNA t_ref_count_DNA i_tumor_f_DNA  \\\n",
       "0       ...         40.720366            18.0            78.0      0.170213   \n",
       "1       ...          6.309237             3.0            37.0      0.075000   \n",
       "2       ...          9.656761             4.0            33.0      0.111111   \n",
       "3       ...         16.651170             8.0            56.0      0.112903   \n",
       "4       ...         36.305156            14.0            56.0      0.188406   \n",
       "...     ...               ...             ...             ...           ...   \n",
       "968891  ...          7.264761             3.0            21.0      0.130435   \n",
       "968892  ...          6.362598             3.0            17.0      0.150000   \n",
       "968893  ...         60.755127            29.0           137.0      0.163522   \n",
       "968894  ...         74.950507            30.0            80.0      0.261682   \n",
       "968895  ...        108.299788            39.0            76.0      0.330357   \n",
       "\n",
       "        Appears_in_rna Cancer_type  Left_flank_base  Right_flank_base  \\\n",
       "0                    0        BLCA                T                 T   \n",
       "1                    1        BLCA                G                 A   \n",
       "2                    0        BLCA                C                 A   \n",
       "3                    1        BLCA                C                 C   \n",
       "4                    1        BLCA                T                 A   \n",
       "...                ...         ...              ...               ...   \n",
       "968891               0        STAD                T                 G   \n",
       "968892               1        STAD                T                 G   \n",
       "968893               0        STAD                T                 C   \n",
       "968894               0        STAD                A                 G   \n",
       "968895               0        STAD                C                 G   \n",
       "\n",
       "        Entrez_Gene_Id   Expression  \n",
       "0                23268   893.592239  \n",
       "1                57715   162.400803  \n",
       "2                81621    61.660977  \n",
       "3                79803   566.412410  \n",
       "4                 4791  1578.509691  \n",
       "...                ...          ...  \n",
       "968891           79776   255.884184  \n",
       "968892            9830  1640.420685  \n",
       "968893          138474    40.830964  \n",
       "968894          203447    82.048767  \n",
       "968895          139378     6.223744  \n",
       "\n",
       "[968896 rows x 29 columns]"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df.merge(expression_df,\n",
    "                left_on = ['Entrez_Gene_Id_DNA', 'Cancer_type'],\n",
    "                right_on = ['Entrez_Gene_Id', 'Cancer_type'], how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "75988c87-7a31-45b7-9f57-e48a20bdc6ab",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol_DNA</th>\n",
       "      <th>Entrez_Gene_Id_DNA</th>\n",
       "      <th>Chromosome_DNA</th>\n",
       "      <th>Start_position_DNA</th>\n",
       "      <th>Variant_Classification_DNA</th>\n",
       "      <th>Reference_Allele_DNA</th>\n",
       "      <th>Tumor_Seq_Allele2_DNA</th>\n",
       "      <th>Tumor_Sample_Barcode_DNA</th>\n",
       "      <th>Transcript_Exon_DNA</th>\n",
       "      <th>Transcript_Position_DNA</th>\n",
       "      <th>...</th>\n",
       "      <th>t_alt_count_DNA</th>\n",
       "      <th>t_ref_count_DNA</th>\n",
       "      <th>i_tumor_f_DNA</th>\n",
       "      <th>Appears_in_rna</th>\n",
       "      <th>Cancer_type</th>\n",
       "      <th>Left_flank_base</th>\n",
       "      <th>Right_flank_base</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Expression</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DNMBP</td>\n",
       "      <td>23268</td>\n",
       "      <td>10</td>\n",
       "      <td>101715548.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1774.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>0.170213</td>\n",
       "      <td>0</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>A[G&gt;A]A</td>\n",
       "      <td>23268</td>\n",
       "      <td>893.592239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SEMA4G</td>\n",
       "      <td>57715</td>\n",
       "      <td>10</td>\n",
       "      <td>102738692.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1103.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>0.075000</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>T[C&gt;G]C</td>\n",
       "      <td>57715</td>\n",
       "      <td>162.400803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KAZALD1</td>\n",
       "      <td>81621</td>\n",
       "      <td>10</td>\n",
       "      <td>102822569.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>2.0</td>\n",
       "      <td>546.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>0.111111</td>\n",
       "      <td>0</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>T[C&gt;T]G</td>\n",
       "      <td>81621</td>\n",
       "      <td>61.660977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HPS6</td>\n",
       "      <td>79803</td>\n",
       "      <td>10</td>\n",
       "      <td>103826020.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>874.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.112903</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>G[G&gt;A]G</td>\n",
       "      <td>79803</td>\n",
       "      <td>566.412410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NFKB2</td>\n",
       "      <td>4791</td>\n",
       "      <td>10</td>\n",
       "      <td>104160055.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1855.0</td>\n",
       "      <td>...</td>\n",
       "      <td>14.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>0.188406</td>\n",
       "      <td>1</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>T[C&gt;G]A</td>\n",
       "      <td>4791</td>\n",
       "      <td>1578.509691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1032202</th>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>79776</td>\n",
       "      <td>8</td>\n",
       "      <td>77763180.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>10.0</td>\n",
       "      <td>4471.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.130435</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>C[C&gt;A]A</td>\n",
       "      <td>79776</td>\n",
       "      <td>255.884184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1032203</th>\n",
       "      <td>TRIM14</td>\n",
       "      <td>9830</td>\n",
       "      <td>9</td>\n",
       "      <td>100881456.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>1</td>\n",
       "      <td>STAD</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>C[G&gt;C]A</td>\n",
       "      <td>9830</td>\n",
       "      <td>1640.420685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1032204</th>\n",
       "      <td>TAF1L</td>\n",
       "      <td>138474</td>\n",
       "      <td>9</td>\n",
       "      <td>32631933.0</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3734.0</td>\n",
       "      <td>...</td>\n",
       "      <td>29.0</td>\n",
       "      <td>137.0</td>\n",
       "      <td>0.163522</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>G[C&gt;T]A</td>\n",
       "      <td>138474</td>\n",
       "      <td>40.830964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1032205</th>\n",
       "      <td>NRK</td>\n",
       "      <td>203447</td>\n",
       "      <td>X</td>\n",
       "      <td>105179181.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>21.0</td>\n",
       "      <td>3822.0</td>\n",
       "      <td>...</td>\n",
       "      <td>30.0</td>\n",
       "      <td>80.0</td>\n",
       "      <td>0.261682</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>C[T&gt;G]T</td>\n",
       "      <td>203447</td>\n",
       "      <td>82.048767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1032206</th>\n",
       "      <td>GPR112</td>\n",
       "      <td>139378</td>\n",
       "      <td>X</td>\n",
       "      <td>135496331.0</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>25.0</td>\n",
       "      <td>9341.0</td>\n",
       "      <td>...</td>\n",
       "      <td>39.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>0.330357</td>\n",
       "      <td>0</td>\n",
       "      <td>STAD</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C[C&gt;T]G</td>\n",
       "      <td>139378</td>\n",
       "      <td>6.223744</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1032207 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Hugo_Symbol_DNA Entrez_Gene_Id_DNA Chromosome_DNA  Start_position_DNA  \\\n",
       "0                 DNMBP              23268             10         101715548.0   \n",
       "1                SEMA4G              57715             10         102738692.0   \n",
       "2               KAZALD1              81621             10         102822569.0   \n",
       "3                  HPS6              79803             10         103826020.0   \n",
       "4                 NFKB2               4791             10         104160055.0   \n",
       "...                 ...                ...            ...                 ...   \n",
       "1032202           ZFHX4              79776              8          77763180.0   \n",
       "1032203          TRIM14               9830              9         100881456.0   \n",
       "1032204           TAF1L             138474              9          32631933.0   \n",
       "1032205             NRK             203447              X         105179181.0   \n",
       "1032206          GPR112             139378              X         135496331.0   \n",
       "\n",
       "        Variant_Classification_DNA Reference_Allele_DNA Tumor_Seq_Allele2_DNA  \\\n",
       "0                           Silent                    C                     T   \n",
       "1                Missense_Mutation                    G                     C   \n",
       "2                Missense_Mutation                    G                     A   \n",
       "3                           Silent                    C                     T   \n",
       "4                           Silent                    G                     C   \n",
       "...                            ...                  ...                   ...   \n",
       "1032202                     Silent                    G                     T   \n",
       "1032203                     Silent                    C                     G   \n",
       "1032204                     Silent                    G                     A   \n",
       "1032205          Missense_Mutation                    A                     C   \n",
       "1032206          Missense_Mutation                    G                     A   \n",
       "\n",
       "        Tumor_Sample_Barcode_DNA  Transcript_Exon_DNA  \\\n",
       "0                BLCA-2F-A9KO-TP                  4.0   \n",
       "1                BLCA-2F-A9KO-TP                  7.0   \n",
       "2                BLCA-2F-A9KO-TP                  2.0   \n",
       "3                BLCA-2F-A9KO-TP                  1.0   \n",
       "4                BLCA-2F-A9KO-TP                 16.0   \n",
       "...                          ...                  ...   \n",
       "1032202          STAD-ZQ-A9CR-TP                 10.0   \n",
       "1032203          STAD-ZQ-A9CR-TP                  1.0   \n",
       "1032204          STAD-ZQ-A9CR-TP                  1.0   \n",
       "1032205          STAD-ZQ-A9CR-TP                 21.0   \n",
       "1032206          STAD-ZQ-A9CR-TP                 25.0   \n",
       "\n",
       "         Transcript_Position_DNA  ... t_alt_count_DNA t_ref_count_DNA  \\\n",
       "0                         1774.0  ...            18.0            78.0   \n",
       "1                         1103.0  ...             3.0            37.0   \n",
       "2                          546.0  ...             4.0            33.0   \n",
       "3                          874.0  ...             8.0            56.0   \n",
       "4                         1855.0  ...            14.0            56.0   \n",
       "...                          ...  ...             ...             ...   \n",
       "1032202                   4471.0  ...             3.0            21.0   \n",
       "1032203                     24.0  ...             3.0            17.0   \n",
       "1032204                   3734.0  ...            29.0           137.0   \n",
       "1032205                   3822.0  ...            30.0            80.0   \n",
       "1032206                   9341.0  ...            39.0            76.0   \n",
       "\n",
       "        i_tumor_f_DNA Appears_in_rna  Cancer_type Left_flank_base  \\\n",
       "0            0.170213              0         BLCA               T   \n",
       "1            0.075000              1         BLCA               G   \n",
       "2            0.111111              0         BLCA               C   \n",
       "3            0.112903              1         BLCA               C   \n",
       "4            0.188406              1         BLCA               T   \n",
       "...               ...            ...          ...             ...   \n",
       "1032202      0.130435              0         STAD               T   \n",
       "1032203      0.150000              1         STAD               T   \n",
       "1032204      0.163522              0         STAD               T   \n",
       "1032205      0.261682              0         STAD               A   \n",
       "1032206      0.330357              0         STAD               C   \n",
       "\n",
       "         Right_flank_base  Mutation  Entrez_Gene_Id   Expression  \n",
       "0                       T   A[G>A]A           23268   893.592239  \n",
       "1                       A   T[C>G]C           57715   162.400803  \n",
       "2                       A   T[C>T]G           81621    61.660977  \n",
       "3                       C   G[G>A]G           79803   566.412410  \n",
       "4                       A   T[C>G]A            4791  1578.509691  \n",
       "...                   ...       ...             ...          ...  \n",
       "1032202                 G   C[C>A]A           79776   255.884184  \n",
       "1032203                 G   C[G>C]A            9830  1640.420685  \n",
       "1032204                 C   G[C>T]A          138474    40.830964  \n",
       "1032205                 G   C[T>G]T          203447    82.048767  \n",
       "1032206                 G   C[C>T]G          139378     6.223744  \n",
       "\n",
       "[1032207 rows x 30 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df.merge(expression_df,\n",
    "                left_on = ['Entrez_Gene_Id_DNA', 'Cancer_type'],\n",
    "                right_on = ['Entrez_Gene_Id', 'Cancer_type'], how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "43bf855f-8442-423a-8c99-d9b7a9f596ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_index = merged_df.index\n",
    "merged_df = merged_df.merge(expression_df,\n",
    "                left_on = ['Entrez_Gene_Id_DNA', 'Cancer_type'],\n",
    "                right_on = ['Entrez_Gene_Id', 'Cancer_type'], how = 'left')\n",
    "merged_df.index = df_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1e3b6211-db84-443a-8a8a-4064ed1de49e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10_101715548_T    central_nervous_system(1)|cervix(4)|endometriu...\n",
       "10_102738692_C    breast(3)|cervix(1)|endometrium(2)|kidney(1)|l...\n",
       "10_102822569_A                  endometrium(1)|ovary(1)|prostate(2)\n",
       "10_103826020_T    endometrium(1)|kidney(1)|large_intestine(4)|lu...\n",
       "10_104160055_C    NS(1)|endometrium(6)|haematopoietic_and_lympho...\n",
       "                                        ...                        \n",
       "8_77763180_T      NS(3)|breast(11)|central_nervous_system(1)|end...\n",
       "9_100881456_G     central_nervous_system(1)|cervix(1)|endometriu...\n",
       "9_32631933_A      breast(6)|central_nervous_system(4)|endometriu...\n",
       "X_105179181_C     breast(8)|central_nervous_system(1)|endometriu...\n",
       "X_135496331_A     NS(2)|breast(13)|central_nervous_system(2)|cer...\n",
       "Name: COSMIC_tissue_types_affected_DNA, Length: 968896, dtype: object"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df['COSMIC_tissue_types_affected_DNA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "f10deab3-1334-49fe-ad62-8b7a8302547b",
   "metadata": {},
   "outputs": [],
   "source": [
    "tissues_df = pd.DataFrame(tissues_dict).fillna(0).astype(int)\n",
    "# categorical_df = merged_df.merge(tissues_df, left_index=True, right_index=True, how='left')\n",
    "\n",
    "numerical_df = pd.get_dummies(merged_df, columns=['Chromosome_DNA', 'Variant_Classification_DNA','Reference_Allele_DNA','Tumor_Seq_Allele2_DNA', 'Left_flank_base', 'Right_flank_base','Cancer_type'])\n",
    "numerical_df = numerical_df.merge(tissues_df, left_index=True, right_index=True, how='left')\n",
    "numerical_df['Cancer_type'] = numerical_df['Tumor_Sample_Barcode_DNA'].str.split('-').str[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "ce964b78-a4ba-457a-9229-ce4d3858c240",
   "metadata": {},
   "outputs": [],
   "source": [
    "drop_columns = ['Hugo_Symbol_DNA', 'ref_context_DNA', 'Transcript_Exon_DNA', 'cDNA_Change_DNA', 'Codon_Change_DNA', 'Protein_Change_DNA', 'COSMIC_tissue_types_affected_DNA']\n",
    "# categorical_df = categorical_df.drop(drop_columns, axis = 1)\n",
    "numerical_df = numerical_df.drop(drop_columns, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0a0a4b78-63f1-4c94-bb9f-5d5786e135b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "numerical_df = numerical_df.fillna(0)\n",
    "# categorical_df = categorical_df.fillna(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "40efc389-3501-4586-b666-f2d674971f91",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('../Output/Numerical_input.csv', index_col=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "144168e3-0ddf-4ede-a3ec-01b08627c3e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df[['Entrez_Gene_Id_DNA', 'Tumor_Sample_Barcode_DNA', 'Appears_in_rna', 'Cancer_type']].to_csv('../Output/Only_expression.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "fcb7e4b8-f100-4a70-af92-dbeeca9eca22",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Cancer_type\n",
       "SKCM    369099\n",
       "LUAD    171793\n",
       "BLCA    126024\n",
       "STAD    125095\n",
       "COAD    108830\n",
       "HNSC     61346\n",
       "PAAD      6709\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Cancer_type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "2a9b772f-9a8b-485b-8ddf-707c8b3ae40e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# categorical_df.to_csv('../Output/Categorical_input.csv')\n",
    "numerical_df.to_csv('../Output/Numerical_input.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "28625498-ff46-49c3-8a79-7a9131c1ccab",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample = expression_df['sample']\n",
    "cancer = expression_df['Cancer_type']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "id": "73b2436e-c8b6-434e-a6f0-974ca64da83e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Hybridization REF\n",
       "100130426       0.084443\n",
       "100133144      14.140035\n",
       "100134869      14.523726\n",
       "10357         152.733826\n",
       "10431         962.013245\n",
       "                ...     \n",
       "7791         5391.243028\n",
       "23140        1336.679062\n",
       "26009         726.402235\n",
       "387590         94.335828\n",
       "389932          0.199441\n",
       "Length: 20531, dtype: float64"
      ]
     },
     "execution_count": 325,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_df = expression_df.drop(['sample', 'Cancer_type'], axis = 1).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "6d06c852-593d-4242-acec-590c63b39ff4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Hybridization REF</th>\n",
       "      <th>100130426</th>\n",
       "      <th>100133144</th>\n",
       "      <th>100134869</th>\n",
       "      <th>10357</th>\n",
       "      <th>10431</th>\n",
       "      <th>136542</th>\n",
       "      <th>155060</th>\n",
       "      <th>26823</th>\n",
       "      <th>280660</th>\n",
       "      <th>317712</th>\n",
       "      <th>...</th>\n",
       "      <th>79364</th>\n",
       "      <th>440590</th>\n",
       "      <th>79699</th>\n",
       "      <th>7791</th>\n",
       "      <th>23140</th>\n",
       "      <th>26009</th>\n",
       "      <th>387590</th>\n",
       "      <th>389932</th>\n",
       "      <th>Cancer_type</th>\n",
       "      <th>sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TCGA-2F-A9KO-01A-11R-A38B-07</th>\n",
       "      <td>0.0000</td>\n",
       "      <td>20.4373</td>\n",
       "      <td>37.8717</td>\n",
       "      <td>123.0904</td>\n",
       "      <td>702.0408</td>\n",
       "      <td>0.0</td>\n",
       "      <td>406.9971</td>\n",
       "      <td>0.5831</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1450.7289</td>\n",
       "      <td>5.8309</td>\n",
       "      <td>697.9592</td>\n",
       "      <td>4262.3907</td>\n",
       "      <td>1787.7551</td>\n",
       "      <td>572.5948</td>\n",
       "      <td>8.7464</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>2F-A9KO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-2F-A9KP-01A-11R-A38B-07</th>\n",
       "      <td>0.0000</td>\n",
       "      <td>16.1382</td>\n",
       "      <td>12.5759</td>\n",
       "      <td>137.8860</td>\n",
       "      <td>882.2305</td>\n",
       "      <td>0.0</td>\n",
       "      <td>182.2722</td>\n",
       "      <td>0.8323</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1420.7241</td>\n",
       "      <td>17.0620</td>\n",
       "      <td>1526.4253</td>\n",
       "      <td>3105.2851</td>\n",
       "      <td>1467.7486</td>\n",
       "      <td>860.5909</td>\n",
       "      <td>2.9130</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>2F-A9KP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-2F-A9KQ-01A-11R-A38B-07</th>\n",
       "      <td>0.0000</td>\n",
       "      <td>13.3333</td>\n",
       "      <td>10.7742</td>\n",
       "      <td>104.6778</td>\n",
       "      <td>954.1029</td>\n",
       "      <td>0.0</td>\n",
       "      <td>224.8493</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>873.4353</td>\n",
       "      <td>1.8544</td>\n",
       "      <td>1571.1637</td>\n",
       "      <td>3275.3825</td>\n",
       "      <td>750.5795</td>\n",
       "      <td>621.2332</td>\n",
       "      <td>3.2452</td>\n",
       "      <td>0.4636</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>2F-A9KQ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-2F-A9KR-01A-11R-A38B-07</th>\n",
       "      <td>0.0000</td>\n",
       "      <td>15.3523</td>\n",
       "      <td>42.5810</td>\n",
       "      <td>146.4527</td>\n",
       "      <td>487.2101</td>\n",
       "      <td>0.0</td>\n",
       "      <td>472.9642</td>\n",
       "      <td>2.3743</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1924.1476</td>\n",
       "      <td>1.8995</td>\n",
       "      <td>748.8600</td>\n",
       "      <td>4465.6181</td>\n",
       "      <td>910.7885</td>\n",
       "      <td>935.4815</td>\n",
       "      <td>366.1199</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>2F-A9KR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-2F-A9KT-01A-11R-A38B-07</th>\n",
       "      <td>0.0000</td>\n",
       "      <td>14.0136</td>\n",
       "      <td>17.6427</td>\n",
       "      <td>142.9621</td>\n",
       "      <td>954.7767</td>\n",
       "      <td>0.0</td>\n",
       "      <td>234.5958</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>938.3833</td>\n",
       "      <td>6.7835</td>\n",
       "      <td>878.4624</td>\n",
       "      <td>3449.4064</td>\n",
       "      <td>850.1979</td>\n",
       "      <td>630.8649</td>\n",
       "      <td>10.1752</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>BLCA</td>\n",
       "      <td>2F-A9KT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-VQ-AA6I-01A-11R-A414-31</th>\n",
       "      <td>0.0000</td>\n",
       "      <td>35.0189</td>\n",
       "      <td>30.1605</td>\n",
       "      <td>21.6463</td>\n",
       "      <td>519.9928</td>\n",
       "      <td>0.0</td>\n",
       "      <td>72.8760</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1032.0486</td>\n",
       "      <td>0.9621</td>\n",
       "      <td>975.7681</td>\n",
       "      <td>4006.9749</td>\n",
       "      <td>1904.8764</td>\n",
       "      <td>719.3795</td>\n",
       "      <td>37.0393</td>\n",
       "      <td>0.4810</td>\n",
       "      <td>STAD</td>\n",
       "      <td>VQ-AA6I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-VQ-AA6J-01A-11R-A414-31</th>\n",
       "      <td>0.2466</td>\n",
       "      <td>27.6770</td>\n",
       "      <td>19.2494</td>\n",
       "      <td>35.6665</td>\n",
       "      <td>709.6427</td>\n",
       "      <td>0.0</td>\n",
       "      <td>94.3317</td>\n",
       "      <td>0.4788</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1078.5898</td>\n",
       "      <td>4.5490</td>\n",
       "      <td>663.1951</td>\n",
       "      <td>4794.6370</td>\n",
       "      <td>1623.9899</td>\n",
       "      <td>1091.2791</td>\n",
       "      <td>29.4487</td>\n",
       "      <td>0.7183</td>\n",
       "      <td>STAD</td>\n",
       "      <td>VQ-AA6J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-VQ-AA6K-01A-11R-A414-31</th>\n",
       "      <td>0.4134</td>\n",
       "      <td>28.6855</td>\n",
       "      <td>21.1378</td>\n",
       "      <td>11.5371</td>\n",
       "      <td>702.4735</td>\n",
       "      <td>0.0</td>\n",
       "      <td>221.5548</td>\n",
       "      <td>0.7067</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1719.7880</td>\n",
       "      <td>3.1802</td>\n",
       "      <td>744.8763</td>\n",
       "      <td>4516.2544</td>\n",
       "      <td>2195.4064</td>\n",
       "      <td>546.6431</td>\n",
       "      <td>12.3675</td>\n",
       "      <td>1.0601</td>\n",
       "      <td>STAD</td>\n",
       "      <td>VQ-AA6K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-ZA-A8F6-01A-23R-A36D-31</th>\n",
       "      <td>0.0000</td>\n",
       "      <td>29.3939</td>\n",
       "      <td>15.4703</td>\n",
       "      <td>22.0386</td>\n",
       "      <td>561.9835</td>\n",
       "      <td>0.0</td>\n",
       "      <td>166.8634</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>915.7812</td>\n",
       "      <td>2.3613</td>\n",
       "      <td>1291.2239</td>\n",
       "      <td>10290.4368</td>\n",
       "      <td>2792.2078</td>\n",
       "      <td>756.0016</td>\n",
       "      <td>138.5281</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>STAD</td>\n",
       "      <td>ZA-A8F6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA-ZQ-A9CR-01A-11R-A39E-31</th>\n",
       "      <td>0.5668</td>\n",
       "      <td>19.8016</td>\n",
       "      <td>8.5379</td>\n",
       "      <td>29.3701</td>\n",
       "      <td>1874.0178</td>\n",
       "      <td>0.0</td>\n",
       "      <td>70.0760</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>740.9507</td>\n",
       "      <td>3.0916</td>\n",
       "      <td>837.8204</td>\n",
       "      <td>6423.2900</td>\n",
       "      <td>956.8466</td>\n",
       "      <td>710.0348</td>\n",
       "      <td>16.4885</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>STAD</td>\n",
       "      <td>ZQ-A9CR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2866 rows × 20533 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Hybridization REF             100130426  100133144  100134869     10357  \\\n",
       "TCGA-2F-A9KO-01A-11R-A38B-07     0.0000    20.4373    37.8717  123.0904   \n",
       "TCGA-2F-A9KP-01A-11R-A38B-07     0.0000    16.1382    12.5759  137.8860   \n",
       "TCGA-2F-A9KQ-01A-11R-A38B-07     0.0000    13.3333    10.7742  104.6778   \n",
       "TCGA-2F-A9KR-01A-11R-A38B-07     0.0000    15.3523    42.5810  146.4527   \n",
       "TCGA-2F-A9KT-01A-11R-A38B-07     0.0000    14.0136    17.6427  142.9621   \n",
       "...                                 ...        ...        ...       ...   \n",
       "TCGA-VQ-AA6I-01A-11R-A414-31     0.0000    35.0189    30.1605   21.6463   \n",
       "TCGA-VQ-AA6J-01A-11R-A414-31     0.2466    27.6770    19.2494   35.6665   \n",
       "TCGA-VQ-AA6K-01A-11R-A414-31     0.4134    28.6855    21.1378   11.5371   \n",
       "TCGA-ZA-A8F6-01A-23R-A36D-31     0.0000    29.3939    15.4703   22.0386   \n",
       "TCGA-ZQ-A9CR-01A-11R-A39E-31     0.5668    19.8016     8.5379   29.3701   \n",
       "\n",
       "Hybridization REF                 10431  136542    155060   26823  280660  \\\n",
       "TCGA-2F-A9KO-01A-11R-A38B-07   702.0408     0.0  406.9971  0.5831     0.0   \n",
       "TCGA-2F-A9KP-01A-11R-A38B-07   882.2305     0.0  182.2722  0.8323     0.0   \n",
       "TCGA-2F-A9KQ-01A-11R-A38B-07   954.1029     0.0  224.8493  0.0000     0.0   \n",
       "TCGA-2F-A9KR-01A-11R-A38B-07   487.2101     0.0  472.9642  2.3743     0.0   \n",
       "TCGA-2F-A9KT-01A-11R-A38B-07   954.7767     0.0  234.5958  0.0000     0.0   \n",
       "...                                 ...     ...       ...     ...     ...   \n",
       "TCGA-VQ-AA6I-01A-11R-A414-31   519.9928     0.0   72.8760  0.0000     0.0   \n",
       "TCGA-VQ-AA6J-01A-11R-A414-31   709.6427     0.0   94.3317  0.4788     0.0   \n",
       "TCGA-VQ-AA6K-01A-11R-A414-31   702.4735     0.0  221.5548  0.7067     0.0   \n",
       "TCGA-ZA-A8F6-01A-23R-A36D-31   561.9835     0.0  166.8634  0.0000     0.0   \n",
       "TCGA-ZQ-A9CR-01A-11R-A39E-31  1874.0178     0.0   70.0760  0.0000     0.0   \n",
       "\n",
       "Hybridization REF             317712  ...      79364   440590      79699  \\\n",
       "TCGA-2F-A9KO-01A-11R-A38B-07     0.0  ...  1450.7289   5.8309   697.9592   \n",
       "TCGA-2F-A9KP-01A-11R-A38B-07     0.0  ...  1420.7241  17.0620  1526.4253   \n",
       "TCGA-2F-A9KQ-01A-11R-A38B-07     0.0  ...   873.4353   1.8544  1571.1637   \n",
       "TCGA-2F-A9KR-01A-11R-A38B-07     0.0  ...  1924.1476   1.8995   748.8600   \n",
       "TCGA-2F-A9KT-01A-11R-A38B-07     0.0  ...   938.3833   6.7835   878.4624   \n",
       "...                              ...  ...        ...      ...        ...   \n",
       "TCGA-VQ-AA6I-01A-11R-A414-31     0.0  ...  1032.0486   0.9621   975.7681   \n",
       "TCGA-VQ-AA6J-01A-11R-A414-31     0.0  ...  1078.5898   4.5490   663.1951   \n",
       "TCGA-VQ-AA6K-01A-11R-A414-31     0.0  ...  1719.7880   3.1802   744.8763   \n",
       "TCGA-ZA-A8F6-01A-23R-A36D-31     0.0  ...   915.7812   2.3613  1291.2239   \n",
       "TCGA-ZQ-A9CR-01A-11R-A39E-31     0.0  ...   740.9507   3.0916   837.8204   \n",
       "\n",
       "Hybridization REF                   7791      23140      26009    387590  \\\n",
       "TCGA-2F-A9KO-01A-11R-A38B-07   4262.3907  1787.7551   572.5948    8.7464   \n",
       "TCGA-2F-A9KP-01A-11R-A38B-07   3105.2851  1467.7486   860.5909    2.9130   \n",
       "TCGA-2F-A9KQ-01A-11R-A38B-07   3275.3825   750.5795   621.2332    3.2452   \n",
       "TCGA-2F-A9KR-01A-11R-A38B-07   4465.6181   910.7885   935.4815  366.1199   \n",
       "TCGA-2F-A9KT-01A-11R-A38B-07   3449.4064   850.1979   630.8649   10.1752   \n",
       "...                                  ...        ...        ...       ...   \n",
       "TCGA-VQ-AA6I-01A-11R-A414-31   4006.9749  1904.8764   719.3795   37.0393   \n",
       "TCGA-VQ-AA6J-01A-11R-A414-31   4794.6370  1623.9899  1091.2791   29.4487   \n",
       "TCGA-VQ-AA6K-01A-11R-A414-31   4516.2544  2195.4064   546.6431   12.3675   \n",
       "TCGA-ZA-A8F6-01A-23R-A36D-31  10290.4368  2792.2078   756.0016  138.5281   \n",
       "TCGA-ZQ-A9CR-01A-11R-A39E-31   6423.2900   956.8466   710.0348   16.4885   \n",
       "\n",
       "Hybridization REF             389932  Cancer_type   sample  \n",
       "TCGA-2F-A9KO-01A-11R-A38B-07  0.0000         BLCA  2F-A9KO  \n",
       "TCGA-2F-A9KP-01A-11R-A38B-07  0.0000         BLCA  2F-A9KP  \n",
       "TCGA-2F-A9KQ-01A-11R-A38B-07  0.4636         BLCA  2F-A9KQ  \n",
       "TCGA-2F-A9KR-01A-11R-A38B-07  0.0000         BLCA  2F-A9KR  \n",
       "TCGA-2F-A9KT-01A-11R-A38B-07  0.0000         BLCA  2F-A9KT  \n",
       "...                              ...          ...      ...  \n",
       "TCGA-VQ-AA6I-01A-11R-A414-31  0.4810         STAD  VQ-AA6I  \n",
       "TCGA-VQ-AA6J-01A-11R-A414-31  0.7183         STAD  VQ-AA6J  \n",
       "TCGA-VQ-AA6K-01A-11R-A414-31  1.0601         STAD  VQ-AA6K  \n",
       "TCGA-ZA-A8F6-01A-23R-A36D-31  0.0000         STAD  ZA-A8F6  \n",
       "TCGA-ZQ-A9CR-01A-11R-A39E-31  0.0000         STAD  ZQ-A9CR  \n",
       "\n",
       "[2866 rows x 20533 columns]"
      ]
     },
     "execution_count": 292,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge expression data with the cancer type data\n",
    "merged_train_df = dataframe.merge(, left_on=gene_column, right_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "adf54e85-9466-4aa6-8672-8c4e5f52ec52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entrez_Gene_Id_DNA</th>\n",
       "      <th>Start_position_DNA</th>\n",
       "      <th>Tumor_Sample_Barcode_DNA</th>\n",
       "      <th>Transcript_Position_DNA</th>\n",
       "      <th>COSMIC_total_alterations_in_gene_DNA</th>\n",
       "      <th>gc_content_DNA</th>\n",
       "      <th>i_COSMIC_n_overlapping_mutations_DNA</th>\n",
       "      <th>i_init_t_lod_DNA</th>\n",
       "      <th>i_t_lod_fstar_DNA</th>\n",
       "      <th>t_alt_count_DNA</th>\n",
       "      <th>...</th>\n",
       "      <th>fallopian_tube</th>\n",
       "      <th>gastrointestinal_tract_</th>\n",
       "      <th>genital_tract</th>\n",
       "      <th>peritoneum</th>\n",
       "      <th>placenta</th>\n",
       "      <th>vagina</th>\n",
       "      <th>vulva</th>\n",
       "      <th>parathyroid</th>\n",
       "      <th>retroperitoneum</th>\n",
       "      <th>paratesticular_tissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10_101715548_T</th>\n",
       "      <td>23268</td>\n",
       "      <td>101715548.0</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>1774.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>0.493</td>\n",
       "      <td>0.0</td>\n",
       "      <td>31.074207</td>\n",
       "      <td>40.720366</td>\n",
       "      <td>18.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10_102738692_C</th>\n",
       "      <td>57715</td>\n",
       "      <td>102738692.0</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>1103.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.602</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.107030</td>\n",
       "      <td>6.309237</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10_102822569_A</th>\n",
       "      <td>81621</td>\n",
       "      <td>102822569.0</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>546.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.736</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.275782</td>\n",
       "      <td>9.656761</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10_103826020_T</th>\n",
       "      <td>79803</td>\n",
       "      <td>103826020.0</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>874.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.632</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.478583</td>\n",
       "      <td>16.651170</td>\n",
       "      <td>8.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10_104160055_C</th>\n",
       "      <td>4791</td>\n",
       "      <td>104160055.0</td>\n",
       "      <td>BLCA-2F-A9KO-TP</td>\n",
       "      <td>1855.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0.662</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.040048</td>\n",
       "      <td>36.305156</td>\n",
       "      <td>14.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8_77763180_T</th>\n",
       "      <td>79776</td>\n",
       "      <td>77763180.0</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>4471.0</td>\n",
       "      <td>432.0</td>\n",
       "      <td>0.403</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.208521</td>\n",
       "      <td>7.264761</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9_100881456_G</th>\n",
       "      <td>9830</td>\n",
       "      <td>100881456.0</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>24.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.731</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.013856</td>\n",
       "      <td>6.362598</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9_32631933_A</th>\n",
       "      <td>138474</td>\n",
       "      <td>32631933.0</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>3734.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>0.433</td>\n",
       "      <td>0.0</td>\n",
       "      <td>43.645163</td>\n",
       "      <td>60.755127</td>\n",
       "      <td>29.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_105179181_C</th>\n",
       "      <td>203447</td>\n",
       "      <td>105179181.0</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>3822.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>0.378</td>\n",
       "      <td>0.0</td>\n",
       "      <td>69.450299</td>\n",
       "      <td>74.950507</td>\n",
       "      <td>30.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_135496331_A</th>\n",
       "      <td>139378</td>\n",
       "      <td>135496331.0</td>\n",
       "      <td>STAD-ZQ-A9CR-TP</td>\n",
       "      <td>9341.0</td>\n",
       "      <td>199.0</td>\n",
       "      <td>0.353</td>\n",
       "      <td>0.0</td>\n",
       "      <td>105.449508</td>\n",
       "      <td>108.299788</td>\n",
       "      <td>39.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>968896 rows × 114 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Entrez_Gene_Id_DNA  Start_position_DNA  \\\n",
       "10_101715548_T              23268         101715548.0   \n",
       "10_102738692_C              57715         102738692.0   \n",
       "10_102822569_A              81621         102822569.0   \n",
       "10_103826020_T              79803         103826020.0   \n",
       "10_104160055_C               4791         104160055.0   \n",
       "...                           ...                 ...   \n",
       "8_77763180_T                79776          77763180.0   \n",
       "9_100881456_G                9830         100881456.0   \n",
       "9_32631933_A               138474          32631933.0   \n",
       "X_105179181_C              203447         105179181.0   \n",
       "X_135496331_A              139378         135496331.0   \n",
       "\n",
       "               Tumor_Sample_Barcode_DNA  Transcript_Position_DNA  \\\n",
       "10_101715548_T          BLCA-2F-A9KO-TP                   1774.0   \n",
       "10_102738692_C          BLCA-2F-A9KO-TP                   1103.0   \n",
       "10_102822569_A          BLCA-2F-A9KO-TP                    546.0   \n",
       "10_103826020_T          BLCA-2F-A9KO-TP                    874.0   \n",
       "10_104160055_C          BLCA-2F-A9KO-TP                   1855.0   \n",
       "...                                 ...                      ...   \n",
       "8_77763180_T            STAD-ZQ-A9CR-TP                   4471.0   \n",
       "9_100881456_G           STAD-ZQ-A9CR-TP                     24.0   \n",
       "9_32631933_A            STAD-ZQ-A9CR-TP                   3734.0   \n",
       "X_105179181_C           STAD-ZQ-A9CR-TP                   3822.0   \n",
       "X_135496331_A           STAD-ZQ-A9CR-TP                   9341.0   \n",
       "\n",
       "                COSMIC_total_alterations_in_gene_DNA  gc_content_DNA  \\\n",
       "10_101715548_T                                  61.0           0.493   \n",
       "10_102738692_C                                  25.0           0.602   \n",
       "10_102822569_A                                   4.0           0.736   \n",
       "10_103826020_T                                  11.0           0.632   \n",
       "10_104160055_C                                  23.0           0.662   \n",
       "...                                              ...             ...   \n",
       "8_77763180_T                                   432.0           0.403   \n",
       "9_100881456_G                                    9.0           0.731   \n",
       "9_32631933_A                                   159.0           0.433   \n",
       "X_105179181_C                                   76.0           0.378   \n",
       "X_135496331_A                                  199.0           0.353   \n",
       "\n",
       "                i_COSMIC_n_overlapping_mutations_DNA  i_init_t_lod_DNA  \\\n",
       "10_101715548_T                                   0.0         31.074207   \n",
       "10_102738692_C                                   0.0         -1.107030   \n",
       "10_102822569_A                                   0.0          4.275782   \n",
       "10_103826020_T                                   0.0          7.478583   \n",
       "10_104160055_C                                   0.0         30.040048   \n",
       "...                                              ...               ...   \n",
       "8_77763180_T                                     0.0          4.208521   \n",
       "9_100881456_G                                    0.0          4.013856   \n",
       "9_32631933_A                                     0.0         43.645163   \n",
       "X_105179181_C                                    0.0         69.450299   \n",
       "X_135496331_A                                    0.0        105.449508   \n",
       "\n",
       "                i_t_lod_fstar_DNA  t_alt_count_DNA  ...  fallopian_tube  \\\n",
       "10_101715548_T          40.720366             18.0  ...             0.0   \n",
       "10_102738692_C           6.309237              3.0  ...             0.0   \n",
       "10_102822569_A           9.656761              4.0  ...             0.0   \n",
       "10_103826020_T          16.651170              8.0  ...             0.0   \n",
       "10_104160055_C          36.305156             14.0  ...             0.0   \n",
       "...                           ...              ...  ...             ...   \n",
       "8_77763180_T             7.264761              3.0  ...             0.0   \n",
       "9_100881456_G            6.362598              3.0  ...             0.0   \n",
       "9_32631933_A            60.755127             29.0  ...             0.0   \n",
       "X_105179181_C           74.950507             30.0  ...             0.0   \n",
       "X_135496331_A          108.299788             39.0  ...             0.0   \n",
       "\n",
       "                gastrointestinal_tract_  genital_tract  peritoneum  placenta  \\\n",
       "10_101715548_T                      0.0            0.0         0.0       0.0   \n",
       "10_102738692_C                      0.0            0.0         0.0       0.0   \n",
       "10_102822569_A                      0.0            0.0         0.0       0.0   \n",
       "10_103826020_T                      0.0            0.0         0.0       0.0   \n",
       "10_104160055_C                      0.0            0.0         0.0       0.0   \n",
       "...                                 ...            ...         ...       ...   \n",
       "8_77763180_T                        0.0            0.0         0.0       0.0   \n",
       "9_100881456_G                       0.0            0.0         0.0       0.0   \n",
       "9_32631933_A                        0.0            0.0         0.0       0.0   \n",
       "X_105179181_C                       0.0            0.0         0.0       0.0   \n",
       "X_135496331_A                       0.0            0.0         0.0       0.0   \n",
       "\n",
       "                vagina  vulva  parathyroid  retroperitoneum  \\\n",
       "10_101715548_T     0.0    0.0          0.0              0.0   \n",
       "10_102738692_C     0.0    0.0          0.0              0.0   \n",
       "10_102822569_A     0.0    0.0          0.0              0.0   \n",
       "10_103826020_T     0.0    0.0          0.0              0.0   \n",
       "10_104160055_C     0.0    0.0          0.0              0.0   \n",
       "...                ...    ...          ...              ...   \n",
       "8_77763180_T       0.0    0.0          0.0              0.0   \n",
       "9_100881456_G      0.0    0.0          0.0              0.0   \n",
       "9_32631933_A       0.0    0.0          0.0              0.0   \n",
       "X_105179181_C      0.0    0.0          0.0              0.0   \n",
       "X_135496331_A      0.0    0.0          0.0              0.0   \n",
       "\n",
       "                paratesticular_tissues  \n",
       "10_101715548_T                     0.0  \n",
       "10_102738692_C                     0.0  \n",
       "10_102822569_A                     0.0  \n",
       "10_103826020_T                     0.0  \n",
       "10_104160055_C                     0.0  \n",
       "...                                ...  \n",
       "8_77763180_T                       0.0  \n",
       "9_100881456_G                      0.0  \n",
       "9_32631933_A                       0.0  \n",
       "X_105179181_C                      0.0  \n",
       "X_135496331_A                      0.0  \n",
       "\n",
       "[968896 rows x 114 columns]"
      ]
     },
     "execution_count": 291,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_processed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "2d9a742b-4738-474e-ab68-524499cc0270",
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_processed_df.to_csv('../Output/model_input.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "c200622e-59b3-4736-af85-2b10bcb94b81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Cancer_type_BLCA', 'Cancer_type_COAD', 'Cancer_type_HNSC',\n",
       "       'Cancer_type_LUAD', 'Cancer_type_PAAD', 'Cancer_type_SKCM',\n",
       "       'Cancer_type_STAD'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_processed_df.columns[filtered_processed_df.columns.str.startswith('Cancer')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "fa4905e5-6061-4f9f-8777-cb45bd356ce0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Cancer_type_BLCA    128664\n",
       "Cancer_type_COAD    111159\n",
       "Cancer_type_HNSC    304413\n",
       "Cancer_type_PAAD      6902\n",
       "Cancer_type_SKCM    380742\n",
       "Cancer_type_STAD    127777\n",
       "dtype: int64"
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_processed_df[['Cancer_type_BLCA', 'Cancer_type_COAD', 'Cancer_type_HNSC', 'Cancer_type_PAAD', 'Cancer_type_SKCM', 'Cancer_type_STAD']].sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "83e6f3b6-ce1c-4ac5-bff5-fdc8dd8fce76",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cancer_type_BLCA\n",
      "\n",
      "Cross-validation results:\n",
      "   accuracy   roc_auc\n",
      "0  0.696693  0.816575\n",
      "1  0.693545  0.816247\n",
      "2  0.692885  0.817173\n",
      "3  0.689465  0.815174\n",
      "4  0.694000  0.814123\n",
      "\n",
      "Mean Accuracy: 0.6933174833846742\n",
      "Mean ROC AUC: 0.8158583266767158\n",
      "Cancer_type_COAD\n",
      "\n",
      "Cross-validation results:\n",
      "   accuracy   roc_auc\n",
      "0  0.729129  0.873062\n",
      "1  0.720448  0.872773\n",
      "2  0.729219  0.872175\n",
      "3  0.728095  0.870481\n",
      "4  0.730332  0.871023\n",
      "\n",
      "Mean Accuracy: 0.7274444974048133\n",
      "Mean ROC AUC: 0.8719028339889519\n",
      "Cancer_type_HNSC\n",
      "\n",
      "Cross-validation results:\n",
      "   accuracy   roc_auc\n",
      "0  0.936337  0.711680\n",
      "1  0.936518  0.708830\n",
      "2  0.935598  0.707625\n",
      "3  0.936418  0.714505\n",
      "4  0.936829  0.713465\n",
      "\n",
      "Mean Accuracy: 0.936339777102338\n",
      "Mean ROC AUC: 0.7112211190523832\n",
      "Cancer_type_PAAD\n",
      "\n",
      "Cross-validation results:\n",
      "   accuracy   roc_auc\n",
      "0  0.769732  0.827393\n",
      "1  0.758870  0.842416\n",
      "2  0.750000  0.809439\n",
      "3  0.757971  0.820079\n",
      "4  0.777536  0.809907\n",
      "\n",
      "Mean Accuracy: 0.7628219416721762\n",
      "Mean ROC AUC: 0.8218465544119408\n",
      "Cancer_type_SKCM\n",
      "\n",
      "Cross-validation results:\n",
      "   accuracy   roc_auc\n",
      "0  0.792880  0.872218\n",
      "1  0.789781  0.873914\n",
      "2  0.788898  0.871166\n",
      "3  0.788609  0.876012\n",
      "4  0.789174  0.874225\n",
      "\n",
      "Mean Accuracy: 0.7898681970365322\n",
      "Mean ROC AUC: 0.8735068746735488\n",
      "Cancer_type_STAD\n",
      "\n",
      "Cross-validation results:\n",
      "   accuracy   roc_auc\n",
      "0  0.646893  0.836395\n",
      "1  0.650141  0.830547\n",
      "2  0.651379  0.833020\n",
      "3  0.647975  0.833981\n",
      "4  0.645588  0.834330\n",
      "\n",
      "Mean Accuracy: 0.6483952491961289\n",
      "Mean ROC AUC: 0.8336546953104523\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn.model_selection import KFold\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.metrics import accuracy_score, roc_auc_score, confusion_matrix\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "\n",
    "results_dict = {}\n",
    "# Define features (X) and target (y)\n",
    "# cancer_type_df = filtered_processed_df\n",
    "# X = cancer_type_df.drop(['Appears_in_rna'], axis=1)\n",
    "# X = cancer_type_df[['i_init_t_lod_DNA', 'i_t_lod_fstar_DNA', 'Cancer_type_HNSC', 'Left_flank_base_C', 'Left_flank_base_G', 'Left_flank_base_T',\n",
    "#        'Right_flank_base_C', 'Right_flank_base_G', 'Right_flank_base_T', 'Expression']]\n",
    "for cancer_type in ['Cancer_type_BLCA', 'Cancer_type_COAD', 'Cancer_type_HNSC', 'Cancer_type_PAAD', 'Cancer_type_SKCM', 'Cancer_type_STAD']:\n",
    "    cancer_type_df = filtered_processed_df[filtered_processed_df[cancer_type]]\n",
    "    X = cancer_type_df[['Expression']]\n",
    "    # X = cancer_type_df.drop('Appears_in_rna', axis = 1)\n",
    "    y = cancer_type_df['Appears_in_rna']\n",
    "    \n",
    "    # Store feature names\n",
    "    feature_names = X.columns\n",
    "    \n",
    "    # Standardize the feature\n",
    "    scaler = StandardScaler()\n",
    "    X = scaler.fit_transform(X)\n",
    "    \n",
    "    # Set up five-fold cross-validation\n",
    "    kf = KFold(n_splits=5, shuffle=True, random_state=42)\n",
    "    \n",
    "    # Initialize the model\n",
    "    model = LogisticRegression()\n",
    "    \n",
    "    # Cross-validation loop\n",
    "    fold_results = []\n",
    "    top_n = 12  # Number of top coefficients to display\n",
    "    \n",
    "    for train_index, test_index in kf.split(X):\n",
    "        # Split the data into training and testing sets\n",
    "        X_train, X_test = X[train_index], X[test_index]\n",
    "        y_train, y_test = y.iloc[train_index], y.iloc[test_index]\n",
    "        \n",
    "        # Train the model\n",
    "        model.fit(X_train, y_train)\n",
    "        coefficients = model.coef_[0]  # Assuming binary logistic regression (1 class)\n",
    "        \n",
    "        # Get indices of the top N coefficients by absolute value\n",
    "        top_indices = np.argsort(np.abs(coefficients))[-top_n:]\n",
    "        \n",
    "        # # Print the top coefficients and their feature names\n",
    "        # print(f\"Top {top_n} coefficients in this fold:\")\n",
    "        # for idx in reversed(top_indices):  # Reverse to show largest first\n",
    "        #     print(f\"Feature: {feature_names[idx]}, Coefficient = {coefficients[idx]:.4f}\")\n",
    "        \n",
    "        # Make predictions\n",
    "        y_pred = model.predict(X_test)\n",
    "        y_pred_prob = model.predict_proba(X_test)[:, 1]\n",
    "        \n",
    "        # Evaluate the model\n",
    "        accuracy = accuracy_score(y_test, y_pred)\n",
    "        roc_auc = roc_auc_score(y_test, y_pred_prob)\n",
    "        fold_results.append({'accuracy': accuracy, 'roc_auc': roc_auc})\n",
    "        \n",
    "        # Print metrics\n",
    "        # print(f\"Fold ROC AUC: {roc_auc:.4f}, Accuracy: {accuracy:.4f}\")\n",
    "    \n",
    "    # Compile results\n",
    "    results_df = pd.DataFrame(fold_results)\n",
    "    print(cancer_type)\n",
    "    print(\"\\nCross-validation results:\")\n",
    "    print(results_df)\n",
    "    print(\"\\nMean Accuracy:\", results_df['accuracy'].mean())\n",
    "    print(\"Mean ROC AUC:\", results_df['roc_auc'].mean())\n",
    "    results_dict[cancer_type] = results_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "8b0404f2-f2a6-439d-986a-350e9a7c0e48",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Run_Prediction.py --model LogisticRegression --cancer_type all\n",
      "Run_Prediction.py --model LogisticRegression --cancer_type BLCA\n",
      "Run_Prediction.py --model LogisticRegression --cancer_type COAD\n",
      "Run_Prediction.py --model LogisticRegression --cancer_type HNSC\n",
      "Run_Prediction.py --model LogisticRegression --cancer_type LUAD\n"
     ]
    }
   ],
   "source": [
    "import itertools\n",
    "import json\n",
    "\n",
    "# Define models and their relevant hyperparameters\n",
    "models_hyperparameters = {\n",
    "    \"LogisticRegression\": {},  # No hyperparameters to vary\n",
    "    \"RandomForest\": {\"n_estimators\": [50, 100, 200], \"max_depth\": [3, 6, 9]},\n",
    "    \"SVM\": {},  # No tunable hyperparameters provided here\n",
    "    \"XGBoost\": {\"n_estimators\": [50, 100], \"learning_rate\": [0.01, 0.1], \"max_depth\": [3, 6, 9]},\n",
    "    \"NeuralNet_1\": {\"learning_rate\": [0.001, 0.005, 0.01], \"epochs\": [10, 20, 50]},\n",
    "    \"NeuralNet_2\": {\"learning_rate\": [0.001, 0.005, 0.01], \"epochs\": [10, 20, 50]},\n",
    "    \"NeuralNet_3\": {\"learning_rate\": [0.001, 0.005, 0.01], \"epochs\": [10, 20, 50]}\n",
    "}\n",
    "\n",
    "# Define cancer types (including \"all\" for all cancers together)\n",
    "cancer_types = [\"all\"] + list(merged_df['Cancer_type'].unique())\n",
    "\n",
    "# Function to generate all argument combinations\n",
    "def generate_arguments(models_hyperparameters, cancer_types):\n",
    "    all_arguments = []\n",
    "\n",
    "    for model, hyperparams in models_hyperparameters.items():\n",
    "        # For models without specific hyperparameters, just create one argument set per cancer type\n",
    "        if not hyperparams:\n",
    "            for cancer_type in cancer_types:\n",
    "                all_arguments.append({\n",
    "                    \"model\": model,\n",
    "                    \"cancer_type\": cancer_type\n",
    "                })\n",
    "        else:\n",
    "            # Generate all combinations of hyperparameters for the current model\n",
    "            keys, values = zip(*hyperparams.items())\n",
    "            hyperparam_combinations = [dict(zip(keys, v)) for v in itertools.product(*values)]\n",
    "            for cancer_type in cancer_types:\n",
    "                all_arguments.append({\n",
    "                    \"model\": model,\n",
    "                    \"cancer_type\": cancer_type\n",
    "                })\n",
    "            for hyperparam_set in hyperparam_combinations:\n",
    "                for cancer_type in cancer_types:\n",
    "                    argument_set = {\"model\": model, \"cancer_type\": cancer_type}\n",
    "                    argument_set.update(hyperparam_set)  # Add hyperparameters to arguments\n",
    "                    all_arguments.append(argument_set)\n",
    "\n",
    "    return all_arguments\n",
    "\n",
    "# Generate all combinations\n",
    "all_arguments = generate_arguments(models_hyperparameters, cancer_types)\n",
    "\n",
    "# Print or save the commands to run the script with these arguments\n",
    "script_name = \"Run_Prediction.py\"  # Replace with the actual script name\n",
    "commands = []\n",
    "for args in all_arguments:\n",
    "    command = f\"{script_name}\"\n",
    "    for key, value in args.items():\n",
    "        command += f\" --{key} {value}\"\n",
    "    commands.append(command)\n",
    "\n",
    "# Save commands to a text file\n",
    "with open(\"../Script/commands.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(commands))\n",
    "\n",
    "# Print a few commands for verification\n",
    "print(\"\\n\".join(commands[:5]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f6f1467-7429-46f1-878f-b71e3c47ecc1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
